<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.353">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>CoMPASS - R code for real data analysis</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>


<link rel="stylesheet" href="../styles.css">
</head>

<body class="nav-sidebar floating nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="../index.html">
    <span class="navbar-title">CoMPASS</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-tutorial" role="button" data-bs-toggle="dropdown" aria-expanded="false" rel="" target="">
 <span class="menu-text">Tutorial</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-tutorial">    
        <li>
    <a class="dropdown-item" href="../paper_code/data_simulation.html" rel="" target="">
 <span class="dropdown-text">Data simulation</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../paper_code/poweranalysis.html" rel="" target="">
 <span class="dropdown-text">Power analysis &amp; sample size</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="../paper_code/concordance_measurement.html" rel="" target="">
 <span class="dropdown-text">Concordance measurement</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item">
    <a class="nav-link" href="../paper_code/about.html" rel="" target="">
 <span class="menu-text">Paper Code Go-through</span></a>
  </li>  
</ul>
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/bioscinema/concordance" rel="" target=""><i class="bi bi-github" role="img">
</i> 
 <span class="menu-text"></span></a>
  </li>  
</ul>
            <div class="quarto-navbar-tools">
</div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
      <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../paper_code/Real_data.html">R code for real data analysis</a></li></ol></nav>
      <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
      </a>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal sidebar-navigation floating overflow-auto">
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../paper_code/simulation.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">R code for simulation results</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../paper_code/Real_data.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">R code for real data analysis</span></a>
  </div>
</li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#loading-datasets-and-correlation-analysis" id="toc-loading-datasets-and-correlation-analysis" class="nav-link active" data-scroll-target="#loading-datasets-and-correlation-analysis">1. Loading datasets and correlation analysis</a></li>
  <li><a href="#diversity-plot" id="toc-diversity-plot" class="nav-link" data-scroll-target="#diversity-plot">2. Diversity plot</a></li>
  <li><a href="#differential-analysis-and-results-visualization" id="toc-differential-analysis-and-results-visualization" class="nav-link" data-scroll-target="#differential-analysis-and-results-visualization">3. Differential analysis and results visualization</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">R code for real data analysis</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<section id="loading-datasets-and-correlation-analysis" class="level2">
<h2 class="anchored" data-anchor-id="loading-datasets-and-correlation-analysis">1. Loading datasets and correlation analysis</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="do">################################################################################################</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="do">############################# Loading your Phyloseq object #####################################</span></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="do">################################################################################################</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="do">## WGS--------------------------------------------------------</span></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"WGSphylo_w_tree.Rdata"</span>)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>phylo_wgs <span class="ot">=</span> <span class="fu">subset_samples</span>(phy, diagnosis <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"cd"</span>,<span class="st">"uc"</span>))</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a>phylo_wgs <span class="ot">=</span> <span class="fu">subset_taxa</span>(phylo_wgs, kingdom<span class="sc">==</span><span class="st">"Bacteria"</span>)</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter to keep only taxa where the domain is 'Bacteria'</span></span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a>tax_table_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">tax_table</span>(physeq))</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a>bacteria_taxa <span class="ot">&lt;-</span> tax_table_df[tax_table_df[, <span class="st">"kingdom"</span>] <span class="sc">==</span> <span class="st">"Bacteria"</span>, ] <span class="do">## Only keep Bacteria</span></span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a>bacteria_ids <span class="ot">&lt;-</span> <span class="fu">rownames</span>(bacteria_taxa)</span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a>phylo_wgs <span class="ot">&lt;-</span> <span class="fu">prune_taxa</span>(bacteria_ids, physeq)</span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a>tax_table <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">tax_table</span>(phylo_wgs))</span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a>tax_table<span class="sc">$</span>phylum <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"bacteriota$"</span>,<span class="st">"bacteria"</span>, tax_table<span class="sc">$</span>phylum, <span class="at">perl =</span> <span class="cn">TRUE</span>)</span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a>tax_table<span class="sc">$</span>phylum <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"bacterota$"</span>,<span class="st">"bacteria"</span>, tax_table<span class="sc">$</span>phylum, <span class="at">perl =</span> <span class="cn">TRUE</span>)</span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a>tax_table<span class="sc">$</span>phylum <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"etes$"</span>,<span class="st">"ota"</span>, tax_table<span class="sc">$</span>phylum, <span class="at">perl =</span> <span class="cn">TRUE</span>)</span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a>tax_table<span class="sc">$</span>phylum <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"bia$"</span>,<span class="st">"biota"</span>, tax_table<span class="sc">$</span>phylum, <span class="at">perl =</span> <span class="cn">TRUE</span>)</span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a>tax_table<span class="sc">$</span>phylum <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"chaota$"</span>,<span class="st">"chaetota"</span>, tax_table<span class="sc">$</span>phylum, <span class="at">perl =</span> <span class="cn">TRUE</span>)</span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a>tax_table<span class="sc">$</span>phylum <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"Deinococcus-Thermus$"</span>,<span class="st">"Deinococcota"</span>, tax_table<span class="sc">$</span>phylum, <span class="at">perl =</span> <span class="cn">TRUE</span>)</span>
<span id="cb1-22"><a href="#cb1-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-23"><a href="#cb1-23" aria-hidden="true" tabindex="-1"></a>tax_matrix <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(tax_table)</span>
<span id="cb1-24"><a href="#cb1-24" aria-hidden="true" tabindex="-1"></a><span class="fu">tax_table</span>(phylo_wgs) <span class="ot">&lt;-</span> tax_matrix</span>
<span id="cb1-25"><a href="#cb1-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-26"><a href="#cb1-26" aria-hidden="true" tabindex="-1"></a><span class="co"># Now bacteria_phyloseq contains only bacterial data</span></span>
<span id="cb1-27"><a href="#cb1-27" aria-hidden="true" tabindex="-1"></a>physeq.wgs.genus <span class="ot">=</span> <span class="fu">aggregate_taxa</span>(phylo_wgs,<span class="at">level =</span> <span class="st">"genus"</span>)</span>
<span id="cb1-28"><a href="#cb1-28" aria-hidden="true" tabindex="-1"></a>physeq.wgs.phylum <span class="ot">=</span> <span class="fu">aggregate_taxa</span>(phylo_wgs,<span class="at">level =</span> <span class="st">"phylum"</span>)</span>
<span id="cb1-29"><a href="#cb1-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-30"><a href="#cb1-30" aria-hidden="true" tabindex="-1"></a><span class="do">## 16s--------------------------------------------------------</span></span>
<span id="cb1-31"><a href="#cb1-31" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"SixteenSphylo.Rdata"</span>)</span>
<span id="cb1-32"><a href="#cb1-32" aria-hidden="true" tabindex="-1"></a>phylo_16s <span class="ot">=</span> <span class="fu">subset_samples</span>(ps_amp_filter, diagnosis <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"uc"</span>,<span class="st">"cd"</span>))</span>
<span id="cb1-33"><a href="#cb1-33" aria-hidden="true" tabindex="-1"></a>phylo_16s <span class="ot">=</span> <span class="fu">subset_taxa</span>(phylo_16s, Kingdom <span class="sc">==</span><span class="st">"Bacteria"</span>)</span>
<span id="cb1-34"><a href="#cb1-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-35"><a href="#cb1-35" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter to keep only taxa where the domain is 'Bacteria'</span></span>
<span id="cb1-36"><a href="#cb1-36" aria-hidden="true" tabindex="-1"></a>tax_table_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">tax_table</span>(physeq))</span>
<span id="cb1-37"><a href="#cb1-37" aria-hidden="true" tabindex="-1"></a>bacteria_taxa <span class="ot">&lt;-</span> tax_table_df[tax_table_df[, <span class="st">"Domain"</span>] <span class="sc">==</span> <span class="st">"Bacteria"</span>, ] <span class="do">## Only keep Bacteria</span></span>
<span id="cb1-38"><a href="#cb1-38" aria-hidden="true" tabindex="-1"></a>bacteria_ids <span class="ot">&lt;-</span> <span class="fu">rownames</span>(bacteria_taxa)</span>
<span id="cb1-39"><a href="#cb1-39" aria-hidden="true" tabindex="-1"></a>phylo_16s <span class="ot">&lt;-</span> <span class="fu">prune_taxa</span>(bacteria_ids, physeq)</span>
<span id="cb1-40"><a href="#cb1-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-41"><a href="#cb1-41" aria-hidden="true" tabindex="-1"></a>tax_table <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">tax_table</span>(phylo_16s))</span>
<span id="cb1-42"><a href="#cb1-42" aria-hidden="true" tabindex="-1"></a>tax_table<span class="sc">$</span>Phylum <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"bacteriota$"</span>,<span class="st">"bacteria"</span>, tax_table<span class="sc">$</span>Phylum, <span class="at">perl =</span> <span class="cn">TRUE</span>)</span>
<span id="cb1-43"><a href="#cb1-43" aria-hidden="true" tabindex="-1"></a>tax_table<span class="sc">$</span>Phylum <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"bacterota$"</span>,<span class="st">"bacteria"</span>, tax_table<span class="sc">$</span>Phylum, <span class="at">perl =</span> <span class="cn">TRUE</span>)</span>
<span id="cb1-44"><a href="#cb1-44" aria-hidden="true" tabindex="-1"></a>tax_matrix <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(tax_table)</span>
<span id="cb1-45"><a href="#cb1-45" aria-hidden="true" tabindex="-1"></a><span class="fu">tax_table</span>(phylo_16s) <span class="ot">&lt;-</span> tax_matrix</span>
<span id="cb1-46"><a href="#cb1-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-47"><a href="#cb1-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-48"><a href="#cb1-48" aria-hidden="true" tabindex="-1"></a>physeq<span class="fl">.16</span>s.genus <span class="ot">=</span> <span class="fu">aggregate_taxa</span>(phylo_16s,<span class="at">level =</span> <span class="st">"Genus"</span>)</span>
<span id="cb1-49"><a href="#cb1-49" aria-hidden="true" tabindex="-1"></a>physeq<span class="fl">.16</span>s.phylum <span class="ot">=</span> <span class="fu">aggregate_taxa</span>(phylo_16s,<span class="at">level =</span> <span class="st">"Phylum"</span>)</span>
<span id="cb1-50"><a href="#cb1-50" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-51"><a href="#cb1-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-52"><a href="#cb1-52" aria-hidden="true" tabindex="-1"></a><span class="do">## Common Features --------------------------------------------------------</span></span>
<span id="cb1-53"><a href="#cb1-53" aria-hidden="true" tabindex="-1"></a>numbers <span class="ot">=</span> <span class="fu">lapply</span>(<span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">7</span>),<span class="cf">function</span>(level){</span>
<span id="cb1-54"><a href="#cb1-54" aria-hidden="true" tabindex="-1"></a>  n1 <span class="ot">=</span> <span class="fu">length</span>(<span class="fu">na.omit</span>(<span class="fu">unique</span>(phylo_16s<span class="sc">@</span>tax_table[, level])))</span>
<span id="cb1-55"><a href="#cb1-55" aria-hidden="true" tabindex="-1"></a>  n2 <span class="ot">=</span> <span class="fu">length</span>(<span class="fu">na.omit</span>(<span class="fu">unique</span>(phylo_wgs<span class="sc">@</span>tax_table[, level])))</span>
<span id="cb1-56"><a href="#cb1-56" aria-hidden="true" tabindex="-1"></a>  n3 <span class="ot">=</span> <span class="fu">length</span>(<span class="fu">intersect</span>(<span class="fu">na.omit</span>(<span class="fu">unique</span>(phylo_wgs<span class="sc">@</span>tax_table[, level])),</span>
<span id="cb1-57"><a href="#cb1-57" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">na.omit</span>(<span class="fu">unique</span>(phylo_16s<span class="sc">@</span>tax_table[, level]))))</span>
<span id="cb1-58"><a href="#cb1-58" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(<span class="fu">c</span>(n1,n2,n3))</span>
<span id="cb1-59"><a href="#cb1-59" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb1-60"><a href="#cb1-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-61"><a href="#cb1-61" aria-hidden="true" tabindex="-1"></a>numbers <span class="ot">=</span> <span class="fu">data.frame</span>(<span class="fu">do.call</span>(cbind,numbers))</span>
<span id="cb1-62"><a href="#cb1-62" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(numbers) <span class="ot">=</span><span class="fu">c</span>(<span class="st">"Domain"</span> ,<span class="st">"Phylum"</span>,<span class="st">"Class"</span>, <span class="st">"Order"</span>, <span class="st">"Family"</span>, <span class="st">"Genus"</span>,<span class="st">"Species"</span>)</span>
<span id="cb1-63"><a href="#cb1-63" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(numbers) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"16s"</span>,<span class="st">"wgs"</span>,<span class="st">"Overlap"</span>)</span>
<span id="cb1-64"><a href="#cb1-64" aria-hidden="true" tabindex="-1"></a>numbers</span>
<span id="cb1-65"><a href="#cb1-65" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-66"><a href="#cb1-66" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to plot and save Venn</span></span>
<span id="cb1-67"><a href="#cb1-67" aria-hidden="true" tabindex="-1"></a>plot_venn_save <span class="ot">&lt;-</span> <span class="cf">function</span>(level, filename) {</span>
<span id="cb1-68"><a href="#cb1-68" aria-hidden="true" tabindex="-1"></a>  only_16s <span class="ot">&lt;-</span> numbers[<span class="st">"16s"</span>, level] <span class="sc">-</span> numbers[<span class="st">"Overlap"</span>, level]</span>
<span id="cb1-69"><a href="#cb1-69" aria-hidden="true" tabindex="-1"></a>  only_wgs <span class="ot">&lt;-</span> numbers[<span class="st">"wgs"</span>, level] <span class="sc">-</span> numbers[<span class="st">"Overlap"</span>, level]</span>
<span id="cb1-70"><a href="#cb1-70" aria-hidden="true" tabindex="-1"></a>  overlap <span class="ot">&lt;-</span> numbers[<span class="st">"Overlap"</span>, level]</span>
<span id="cb1-71"><a href="#cb1-71" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb1-72"><a href="#cb1-72" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Open PDF file</span></span>
<span id="cb1-73"><a href="#cb1-73" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pdf</span>(<span class="at">file =</span> filename, <span class="at">width =</span> <span class="dv">5</span>, <span class="at">height =</span> <span class="dv">3</span>)  <span class="co"># You can adjust width/height</span></span>
<span id="cb1-74"><a href="#cb1-74" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb1-75"><a href="#cb1-75" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Plot Venn</span></span>
<span id="cb1-76"><a href="#cb1-76" aria-hidden="true" tabindex="-1"></a>  VennDiagram<span class="sc">::</span><span class="fu">draw.pairwise.venn</span>(</span>
<span id="cb1-77"><a href="#cb1-77" aria-hidden="true" tabindex="-1"></a>    <span class="at">area1 =</span> only_16s <span class="sc">+</span> overlap,</span>
<span id="cb1-78"><a href="#cb1-78" aria-hidden="true" tabindex="-1"></a>    <span class="at">area2 =</span> only_wgs <span class="sc">+</span> overlap,</span>
<span id="cb1-79"><a href="#cb1-79" aria-hidden="true" tabindex="-1"></a>    <span class="at">cross.area =</span> overlap,</span>
<span id="cb1-80"><a href="#cb1-80" aria-hidden="true" tabindex="-1"></a>    <span class="at">category =</span> <span class="fu">c</span>(<span class="st">""</span>, <span class="st">""</span>),</span>
<span id="cb1-81"><a href="#cb1-81" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill =</span> <span class="fu">c</span>(<span class="st">"#ffa551"</span>, <span class="st">"#70afdf"</span>),</span>
<span id="cb1-82"><a href="#cb1-82" aria-hidden="true" tabindex="-1"></a>    <span class="at">lty =</span> <span class="st">"blank"</span>,</span>
<span id="cb1-83"><a href="#cb1-83" aria-hidden="true" tabindex="-1"></a>    <span class="at">cex =</span> <span class="dv">3</span>,</span>
<span id="cb1-84"><a href="#cb1-84" aria-hidden="true" tabindex="-1"></a>    <span class="at">cat.cex =</span> <span class="dv">2</span>,</span>
<span id="cb1-85"><a href="#cb1-85" aria-hidden="true" tabindex="-1"></a>    <span class="at">cat.pos =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">20</span>, <span class="dv">20</span>),</span>
<span id="cb1-86"><a href="#cb1-86" aria-hidden="true" tabindex="-1"></a>    <span class="at">cat.dist =</span> <span class="fl">0.05</span></span>
<span id="cb1-87"><a href="#cb1-87" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb1-88"><a href="#cb1-88" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb1-89"><a href="#cb1-89" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Add the label on the left</span></span>
<span id="cb1-90"><a href="#cb1-90" aria-hidden="true" tabindex="-1"></a>  <span class="fu">grid.text</span>(level, <span class="at">x =</span> <span class="fl">0.12</span>, <span class="at">y =</span> <span class="fl">0.95</span>, <span class="at">gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">24</span>, <span class="at">fontface =</span> <span class="st">"bold"</span>))</span>
<span id="cb1-91"><a href="#cb1-91" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb1-92"><a href="#cb1-92" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb1-93"><a href="#cb1-93" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Close and save the PDF</span></span>
<span id="cb1-94"><a href="#cb1-94" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dev.off</span>()</span>
<span id="cb1-95"><a href="#cb1-95" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb1-96"><a href="#cb1-96" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-97"><a href="#cb1-97" aria-hidden="true" tabindex="-1"></a><span class="co"># Example: Save the "Genus" Venn diagram</span></span>
<span id="cb1-98"><a href="#cb1-98" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_venn_save</span>(<span class="st">"Genus"</span>, <span class="st">"Genus_Venn.pdf"</span>)</span>
<span id="cb1-99"><a href="#cb1-99" aria-hidden="true" tabindex="-1"></a><span class="do">###################################################################################################################</span></span>
<span id="cb1-100"><a href="#cb1-100" aria-hidden="true" tabindex="-1"></a><span class="do">################################# Common Genus Feature Correlations ###############################################</span></span>
<span id="cb1-101"><a href="#cb1-101" aria-hidden="true" tabindex="-1"></a><span class="do">###################################################################################################################</span></span>
<span id="cb1-102"><a href="#cb1-102" aria-hidden="true" tabindex="-1"></a><span class="do">## Check if OTU table samples order matches!!!</span></span>
<span id="cb1-103"><a href="#cb1-103" aria-hidden="true" tabindex="-1"></a><span class="fu">all</span>(<span class="fu">colnames</span>(physeq<span class="fl">.16</span>s.genus<span class="sc">@</span>otu_table) <span class="sc">==</span> <span class="fu">colnames</span>(physeq.wgs.genus<span class="sc">@</span>otu_table))</span>
<span id="cb1-104"><a href="#cb1-104" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-105"><a href="#cb1-105" aria-hidden="true" tabindex="-1"></a><span class="do">## Transform to Relative Abundance</span></span>
<span id="cb1-106"><a href="#cb1-106" aria-hidden="true" tabindex="-1"></a>physeq<span class="fl">.16</span>s.genus.rel <span class="ot">=</span> <span class="fu">transform_sample_counts</span>(physeq<span class="fl">.16</span>s.genus, <span class="cf">function</span>(x) x <span class="sc">/</span> <span class="fu">sum</span>(x))</span>
<span id="cb1-107"><a href="#cb1-107" aria-hidden="true" tabindex="-1"></a>physeq.wgs.genus.rel <span class="ot">=</span> <span class="fu">transform_sample_counts</span>(physeq.wgs.genus, <span class="cf">function</span>(x) x <span class="sc">/</span> <span class="fu">sum</span>(x))</span>
<span id="cb1-108"><a href="#cb1-108" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-109"><a href="#cb1-109" aria-hidden="true" tabindex="-1"></a><span class="do">## Common genus features from the most to least abundant</span></span>
<span id="cb1-110"><a href="#cb1-110" aria-hidden="true" tabindex="-1"></a>genus_16s <span class="ot">=</span> <span class="fu">names</span>(<span class="fu">sort</span>(<span class="fu">taxa_sums</span>(physeq<span class="fl">.16</span>s.genus.rel), <span class="cn">TRUE</span>))</span>
<span id="cb1-111"><a href="#cb1-111" aria-hidden="true" tabindex="-1"></a>genus_16s  <span class="ot">=</span> genus_16s[<span class="sc">!</span><span class="fu">grepl</span>(<span class="st">"Unknown"</span>, genus_16s)]</span>
<span id="cb1-112"><a href="#cb1-112" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-113"><a href="#cb1-113" aria-hidden="true" tabindex="-1"></a>genus_wgs <span class="ot">=</span> <span class="fu">names</span>(<span class="fu">sort</span>(<span class="fu">taxa_sums</span>(physeq.wgs.genus.rel), <span class="cn">TRUE</span>))</span>
<span id="cb1-114"><a href="#cb1-114" aria-hidden="true" tabindex="-1"></a>genus_wgs <span class="ot">=</span> genus_wgs[<span class="sc">!</span><span class="fu">grepl</span>(<span class="st">"Unknown"</span>, genus_wgs)]</span>
<span id="cb1-115"><a href="#cb1-115" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-116"><a href="#cb1-116" aria-hidden="true" tabindex="-1"></a>common_genus <span class="ot">=</span> <span class="fu">intersect</span>(genus_16s,genus_wgs)</span>
<span id="cb1-117"><a href="#cb1-117" aria-hidden="true" tabindex="-1"></a>common_genus_df <span class="ot">=</span>  <span class="fu">data.frame</span>(<span class="at">common =</span> common_genus, </span>
<span id="cb1-118"><a href="#cb1-118" aria-hidden="true" tabindex="-1"></a>                              <span class="at">SixS_Rank =</span> <span class="fu">match</span>(common_genus, genus_16s),</span>
<span id="cb1-119"><a href="#cb1-119" aria-hidden="true" tabindex="-1"></a>                              <span class="at">WGS_Rank =</span> <span class="fu">match</span>(common_genus, genus_wgs))</span>
<span id="cb1-120"><a href="#cb1-120" aria-hidden="true" tabindex="-1"></a><span class="co">#write.csv(common_genus_df,file = "Full_common_genus.csv",row.names = FALSE)</span></span>
<span id="cb1-121"><a href="#cb1-121" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-122"><a href="#cb1-122" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-123"><a href="#cb1-123" aria-hidden="true" tabindex="-1"></a><span class="do">## Pearson Correlation ---------------------------------------------------------------------------</span></span>
<span id="cb1-124"><a href="#cb1-124" aria-hidden="true" tabindex="-1"></a>common_genus_cor <span class="ot">=</span> <span class="fu">lapply</span>(common_genus, <span class="cf">function</span>(feature){</span>
<span id="cb1-125"><a href="#cb1-125" aria-hidden="true" tabindex="-1"></a>  rel_abun_16s <span class="ot">=</span> <span class="fu">as.vector</span>(physeq<span class="fl">.16</span>s.genus.rel<span class="sc">@</span>otu_table[feature,])</span>
<span id="cb1-126"><a href="#cb1-126" aria-hidden="true" tabindex="-1"></a>  rel_abun_wgs <span class="ot">=</span> <span class="fu">as.vector</span>(physeq.wgs.genus.rel<span class="sc">@</span>otu_table[feature,])</span>
<span id="cb1-127"><a href="#cb1-127" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cor</span>(rel_abun_16s, rel_abun_wgs)</span>
<span id="cb1-128"><a href="#cb1-128" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb1-129"><a href="#cb1-129" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-130"><a href="#cb1-130" aria-hidden="true" tabindex="-1"></a><span class="do">## Distance Correlation ---------------------------------------------------------------------------</span></span>
<span id="cb1-131"><a href="#cb1-131" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(energy)</span>
<span id="cb1-132"><a href="#cb1-132" aria-hidden="true" tabindex="-1"></a>common_genus_cor <span class="ot">=</span> <span class="fu">lapply</span>(common_genus, <span class="cf">function</span>(feature){</span>
<span id="cb1-133"><a href="#cb1-133" aria-hidden="true" tabindex="-1"></a>  rel_abun_16s <span class="ot">=</span> <span class="fu">as.vector</span>(physeq<span class="fl">.16</span>s.genus.rel<span class="sc">@</span>otu_table[feature,])</span>
<span id="cb1-134"><a href="#cb1-134" aria-hidden="true" tabindex="-1"></a>  rel_abun_wgs <span class="ot">=</span> <span class="fu">as.vector</span>(physeq.wgs.genus.rel<span class="sc">@</span>otu_table[feature,])</span>
<span id="cb1-135"><a href="#cb1-135" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dcor</span>(rel_abun_16s, rel_abun_wgs)</span>
<span id="cb1-136"><a href="#cb1-136" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb1-137"><a href="#cb1-137" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-138"><a href="#cb1-138" aria-hidden="true" tabindex="-1"></a><span class="do">## Correlation Plot ------------------------------------------------------------------------------</span></span>
<span id="cb1-139"><a href="#cb1-139" aria-hidden="true" tabindex="-1"></a>common_genus_cor <span class="ot">=</span> <span class="fu">do.call</span>(rbind,common_genus_cor)</span>
<span id="cb1-140"><a href="#cb1-140" aria-hidden="true" tabindex="-1"></a>common_genus_cor_df <span class="ot">=</span> <span class="fu">data.frame</span>(<span class="at">Genus =</span> common_genus,<span class="at">Corr =</span> common_genus_cor)</span>
<span id="cb1-141"><a href="#cb1-141" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-142"><a href="#cb1-142" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(viridis)</span>
<span id="cb1-143"><a href="#cb1-143" aria-hidden="true" tabindex="-1"></a>color_palette <span class="ot">=</span> <span class="fu">viridis_pal</span>()(<span class="fu">length</span>(<span class="fu">unique</span>(common_genus_cor_df<span class="sc">$</span>Genus)))</span>
<span id="cb1-144"><a href="#cb1-144" aria-hidden="true" tabindex="-1"></a>common_genus_cor_df<span class="sc">$</span>Genus <span class="ot">=</span> <span class="fu">factor</span>(common_genus_cor_df<span class="sc">$</span>Genus, <span class="at">levels =</span> common_genus_cor_df<span class="sc">$</span>Genus)</span>
<span id="cb1-145"><a href="#cb1-145" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-146"><a href="#cb1-146" aria-hidden="true" tabindex="-1"></a>corr_plot <span class="ot">=</span> <span class="fu">ggplot</span>(<span class="at">data=</span>common_genus_cor_df, <span class="fu">aes</span>(<span class="at">x=</span>Genus, <span class="at">y=</span>Corr,<span class="at">fill =</span> Genus)) <span class="sc">+</span></span>
<span id="cb1-147"><a href="#cb1-147" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">"identity"</span>)<span class="sc">+</span></span>
<span id="cb1-148"><a href="#cb1-148" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> color_palette) <span class="sc">+</span>  </span>
<span id="cb1-149"><a href="#cb1-149" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="cn">NULL</span>, <span class="at">y =</span> <span class="st">"Correlation"</span>,<span class="at">title =</span> <span class="st">"Common Genera"</span>)  <span class="sc">+</span></span>
<span id="cb1-150"><a href="#cb1-150" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"right"</span>,<span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(),  </span>
<span id="cb1-151"><a href="#cb1-151" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.x =</span> <span class="fu">element_blank</span>()) </span>
<span id="cb1-152"><a href="#cb1-152" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-153"><a href="#cb1-153" aria-hidden="true" tabindex="-1"></a><span class="co">#ggsave("Real_Plots/genus_dcorr_plot.pdf",corr_plot,height = 5,width = 20)</span></span>
<span id="cb1-154"><a href="#cb1-154" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-155"><a href="#cb1-155" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-156"><a href="#cb1-156" aria-hidden="true" tabindex="-1"></a><span class="do">## Feature-wise scatter plot ---------------------------------------------------------------------</span></span>
<span id="cb1-157"><a href="#cb1-157" aria-hidden="true" tabindex="-1"></a>Most_Abun_Genus <span class="ot">=</span> <span class="fu">as.character</span>(common_genus_cor_df[<span class="dv">1</span><span class="sc">:</span><span class="dv">50</span>,<span class="st">"Genus"</span>])</span>
<span id="cb1-158"><a href="#cb1-158" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-159"><a href="#cb1-159" aria-hidden="true" tabindex="-1"></a>Most_Abun_Genus_Plots <span class="ot">=</span> <span class="fu">lapply</span>(Most_Abun_Genus, <span class="cf">function</span>(feature){</span>
<span id="cb1-160"><a href="#cb1-160" aria-hidden="true" tabindex="-1"></a>  feature_df <span class="ot">=</span> <span class="fu">data.frame</span>(<span class="at">rel_16s =</span> <span class="fu">as.vector</span>(physeq<span class="fl">.16</span>s.genus.rel<span class="sc">@</span>otu_table[feature,]),</span>
<span id="cb1-161"><a href="#cb1-161" aria-hidden="true" tabindex="-1"></a>                          <span class="at">rel_wgs =</span> <span class="fu">as.vector</span>(physeq.wgs.genus.rel<span class="sc">@</span>otu_table[feature,]))</span>
<span id="cb1-162"><a href="#cb1-162" aria-hidden="true" tabindex="-1"></a>  limit <span class="ot">=</span> <span class="fu">max</span>(feature_df)</span>
<span id="cb1-163"><a href="#cb1-163" aria-hidden="true" tabindex="-1"></a>  cor_value <span class="ot">=</span> <span class="fu">round</span>(common_genus_cor_df[<span class="fu">which</span>(common_genus_cor_df<span class="sc">$</span>Genus <span class="sc">==</span> feature),<span class="st">"Corr"</span>],<span class="dv">2</span>)</span>
<span id="cb1-164"><a href="#cb1-164" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(feature_df, <span class="fu">aes</span>(<span class="at">x=</span>rel_16s, <span class="at">y=</span>rel_wgs))  <span class="sc">+</span> </span>
<span id="cb1-165"><a href="#cb1-165" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb1-166"><a href="#cb1-166" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_smooth</span>(<span class="at">method=</span>lm) <span class="sc">+</span></span>
<span id="cb1-167"><a href="#cb1-167" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_classic</span>()<span class="sc">+</span></span>
<span id="cb1-168"><a href="#cb1-168" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">title =</span> <span class="fu">paste0</span>(feature))<span class="sc">+</span></span>
<span id="cb1-169"><a href="#cb1-169" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ylim</span>(<span class="dv">0</span>, limit)<span class="sc">+</span></span>
<span id="cb1-170"><a href="#cb1-170" aria-hidden="true" tabindex="-1"></a>    <span class="fu">xlim</span>(<span class="dv">0</span>, limit)<span class="sc">+</span></span>
<span id="cb1-171"><a href="#cb1-171" aria-hidden="true" tabindex="-1"></a>    <span class="fu">annotate</span>(<span class="st">"text"</span>,<span class="at">x =</span> <span class="dv">0</span>, <span class="at">y =</span> limit,<span class="at">label =</span> <span class="fu">paste0</span>(<span class="st">"r="</span>, cor_value),<span class="at">hjust =</span> <span class="sc">-</span><span class="fl">0.1</span>, <span class="at">vjust =</span> <span class="fl">1.1</span>,<span class="at">size =</span> <span class="dv">5</span>)</span>
<span id="cb1-172"><a href="#cb1-172" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb1-173"><a href="#cb1-173" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-174"><a href="#cb1-174" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb1-175"><a href="#cb1-175" aria-hidden="true" tabindex="-1"></a>Combined_Plots <span class="ot">&lt;-</span> <span class="fu">wrap_plots</span>(Most_Abun_Genus_Plots, <span class="at">ncol =</span> <span class="dv">10</span>)</span>
<span id="cb1-176"><a href="#cb1-176" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-177"><a href="#cb1-177" aria-hidden="true" tabindex="-1"></a><span class="co">#ggsave("Real_Plots/Most_Abun_Genus_Plots.pdf",Combined_Plots,height = 10,width = 20)</span></span>
<span id="cb1-178"><a href="#cb1-178" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-179"><a href="#cb1-179" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-180"><a href="#cb1-180" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-181"><a href="#cb1-181" aria-hidden="true" tabindex="-1"></a><span class="do">###################################################################################################################</span></span>
<span id="cb1-182"><a href="#cb1-182" aria-hidden="true" tabindex="-1"></a><span class="do">################################# Common Phylum Feature Correlations ###############################################</span></span>
<span id="cb1-183"><a href="#cb1-183" aria-hidden="true" tabindex="-1"></a><span class="do">###################################################################################################################</span></span>
<span id="cb1-184"><a href="#cb1-184" aria-hidden="true" tabindex="-1"></a><span class="do">## Check if OTU table samples order matches!!!</span></span>
<span id="cb1-185"><a href="#cb1-185" aria-hidden="true" tabindex="-1"></a><span class="fu">all</span>(<span class="fu">colnames</span>(physeq<span class="fl">.16</span>s.phylum<span class="sc">@</span>otu_table) <span class="sc">==</span> <span class="fu">colnames</span>(physeq.wgs.phylum<span class="sc">@</span>otu_table))</span>
<span id="cb1-186"><a href="#cb1-186" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-187"><a href="#cb1-187" aria-hidden="true" tabindex="-1"></a><span class="do">## Transform to Relative Abundance</span></span>
<span id="cb1-188"><a href="#cb1-188" aria-hidden="true" tabindex="-1"></a>physeq<span class="fl">.16</span>s.phylum.rel <span class="ot">=</span> <span class="fu">transform_sample_counts</span>(physeq<span class="fl">.16</span>s.phylum, <span class="cf">function</span>(x) x <span class="sc">/</span> <span class="fu">sum</span>(x))</span>
<span id="cb1-189"><a href="#cb1-189" aria-hidden="true" tabindex="-1"></a>physeq.wgs.phylum.rel <span class="ot">=</span> <span class="fu">transform_sample_counts</span>(physeq.wgs.phylum, <span class="cf">function</span>(x) x <span class="sc">/</span> <span class="fu">sum</span>(x))</span>
<span id="cb1-190"><a href="#cb1-190" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-191"><a href="#cb1-191" aria-hidden="true" tabindex="-1"></a><span class="do">## Common genus features from the most to least abundant</span></span>
<span id="cb1-192"><a href="#cb1-192" aria-hidden="true" tabindex="-1"></a>phylum_16s <span class="ot">=</span> <span class="fu">names</span>(<span class="fu">sort</span>(<span class="fu">taxa_sums</span>(physeq<span class="fl">.16</span>s.phylum.rel), <span class="cn">TRUE</span>))</span>
<span id="cb1-193"><a href="#cb1-193" aria-hidden="true" tabindex="-1"></a>phylum_16s  <span class="ot">=</span> phylum_16s [<span class="sc">!</span><span class="fu">grepl</span>(<span class="st">"Unknown"</span>, phylum_16s)]</span>
<span id="cb1-194"><a href="#cb1-194" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-195"><a href="#cb1-195" aria-hidden="true" tabindex="-1"></a>phylum_wgs <span class="ot">=</span> <span class="fu">names</span>(<span class="fu">sort</span>(<span class="fu">taxa_sums</span>(physeq.wgs.phylum.rel), <span class="cn">TRUE</span>))</span>
<span id="cb1-196"><a href="#cb1-196" aria-hidden="true" tabindex="-1"></a>phylum_wgs <span class="ot">=</span> phylum_wgs[<span class="sc">!</span><span class="fu">grepl</span>(<span class="st">"Unknown"</span>, phylum_wgs)]</span>
<span id="cb1-197"><a href="#cb1-197" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-198"><a href="#cb1-198" aria-hidden="true" tabindex="-1"></a>common_phylum <span class="ot">=</span> <span class="fu">intersect</span>(phylum_16s,phylum_wgs)</span>
<span id="cb1-199"><a href="#cb1-199" aria-hidden="true" tabindex="-1"></a><span class="co">#save(common_phylum ,file = "Full_common_phylum.Rdata")</span></span>
<span id="cb1-200"><a href="#cb1-200" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-201"><a href="#cb1-201" aria-hidden="true" tabindex="-1"></a><span class="do">## Pearson Correlation ---------------------------------------------------------------------------</span></span>
<span id="cb1-202"><a href="#cb1-202" aria-hidden="true" tabindex="-1"></a>common_phylum_cor <span class="ot">=</span> <span class="fu">lapply</span>(common_phylum, <span class="cf">function</span>(feature){</span>
<span id="cb1-203"><a href="#cb1-203" aria-hidden="true" tabindex="-1"></a>  rel_abun_16s <span class="ot">=</span> <span class="fu">as.vector</span>(physeq<span class="fl">.16</span>s.phylum.rel<span class="sc">@</span>otu_table[feature,])</span>
<span id="cb1-204"><a href="#cb1-204" aria-hidden="true" tabindex="-1"></a>  rel_abun_wgs <span class="ot">=</span> <span class="fu">as.vector</span>(physeq.wgs.phylum.rel<span class="sc">@</span>otu_table[feature,])</span>
<span id="cb1-205"><a href="#cb1-205" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cor</span>(rel_abun_16s, rel_abun_wgs)</span>
<span id="cb1-206"><a href="#cb1-206" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb1-207"><a href="#cb1-207" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-208"><a href="#cb1-208" aria-hidden="true" tabindex="-1"></a><span class="do">## Distance Correlation ---------------------------------------------------------------------------</span></span>
<span id="cb1-209"><a href="#cb1-209" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(energy)</span>
<span id="cb1-210"><a href="#cb1-210" aria-hidden="true" tabindex="-1"></a>common_phylum_cor <span class="ot">=</span> <span class="fu">lapply</span>(common_phylum, <span class="cf">function</span>(feature){</span>
<span id="cb1-211"><a href="#cb1-211" aria-hidden="true" tabindex="-1"></a>  rel_abun_16s <span class="ot">=</span> <span class="fu">as.vector</span>(physeq<span class="fl">.16</span>s.phylum.rel<span class="sc">@</span>otu_table[feature,])</span>
<span id="cb1-212"><a href="#cb1-212" aria-hidden="true" tabindex="-1"></a>  rel_abun_wgs <span class="ot">=</span> <span class="fu">as.vector</span>(physeq.wgs.phylum.rel<span class="sc">@</span>otu_table[feature,])</span>
<span id="cb1-213"><a href="#cb1-213" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dcor</span>(rel_abun_16s, rel_abun_wgs)</span>
<span id="cb1-214"><a href="#cb1-214" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb1-215"><a href="#cb1-215" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-216"><a href="#cb1-216" aria-hidden="true" tabindex="-1"></a><span class="do">## Correlation Plot ------------------------------------------------------------------------------</span></span>
<span id="cb1-217"><a href="#cb1-217" aria-hidden="true" tabindex="-1"></a>common_phylum_cor <span class="ot">=</span> <span class="fu">do.call</span>(rbind,common_phylum_cor)</span>
<span id="cb1-218"><a href="#cb1-218" aria-hidden="true" tabindex="-1"></a>common_phylum_cor_df <span class="ot">=</span> <span class="fu">data.frame</span>(<span class="at">Phylum =</span> common_phylum,<span class="at">Corr =</span> common_phylum_cor)</span>
<span id="cb1-219"><a href="#cb1-219" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-220"><a href="#cb1-220" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(viridis)</span>
<span id="cb1-221"><a href="#cb1-221" aria-hidden="true" tabindex="-1"></a>color_palette <span class="ot">=</span> <span class="fu">viridis_pal</span>()(<span class="fu">length</span>(<span class="fu">unique</span>(common_phylum_cor_df<span class="sc">$</span>Phylum)))</span>
<span id="cb1-222"><a href="#cb1-222" aria-hidden="true" tabindex="-1"></a>common_phylum_cor_df<span class="sc">$</span>Phylum <span class="ot">=</span> <span class="fu">factor</span>(common_phylum_cor_df<span class="sc">$</span>Phylum, <span class="at">levels =</span> common_phylum_cor_df<span class="sc">$</span>Phylum)</span>
<span id="cb1-223"><a href="#cb1-223" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-224"><a href="#cb1-224" aria-hidden="true" tabindex="-1"></a>corr_plot <span class="ot">=</span> <span class="fu">ggplot</span>(<span class="at">data=</span>common_phylum_cor_df, <span class="fu">aes</span>(<span class="at">x=</span>Phylum, <span class="at">y=</span>Corr,<span class="at">fill =</span> Phylum)) <span class="sc">+</span></span>
<span id="cb1-225"><a href="#cb1-225" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">"identity"</span>)<span class="sc">+</span></span>
<span id="cb1-226"><a href="#cb1-226" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> color_palette) <span class="sc">+</span>  </span>
<span id="cb1-227"><a href="#cb1-227" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="cn">NULL</span>, <span class="at">y =</span> <span class="cn">NULL</span>)  <span class="sc">+</span></span>
<span id="cb1-228"><a href="#cb1-228" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylim</span>(<span class="sc">-</span><span class="fl">0.05</span>,<span class="dv">1</span>)<span class="sc">+</span></span>
<span id="cb1-229"><a href="#cb1-229" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()<span class="sc">+</span></span>
<span id="cb1-230"><a href="#cb1-230" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"right"</span>,</span>
<span id="cb1-231"><a href="#cb1-231" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">8</span>),</span>
<span id="cb1-232"><a href="#cb1-232" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.key.size =</span> <span class="fu">unit</span>(<span class="fl">0.3</span>, <span class="st">"cm"</span>),</span>
<span id="cb1-233"><a href="#cb1-233" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(),  </span>
<span id="cb1-234"><a href="#cb1-234" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb1-235"><a href="#cb1-235" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.grid =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb1-236"><a href="#cb1-236" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.line =</span> <span class="fu">element_line</span>(<span class="at">color =</span> <span class="st">"black"</span>)) </span>
<span id="cb1-237"><a href="#cb1-237" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-238"><a href="#cb1-238" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"Real_Plots/phylum_corr_plot.pdf"</span>,corr_plot,<span class="at">height =</span> <span class="dv">2</span>,<span class="at">width =</span> <span class="dv">4</span>)</span>
<span id="cb1-239"><a href="#cb1-239" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-240"><a href="#cb1-240" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-241"><a href="#cb1-241" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-242"><a href="#cb1-242" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-243"><a href="#cb1-243" aria-hidden="true" tabindex="-1"></a><span class="do">## Plot of Most Abundant Features</span></span>
<span id="cb1-244"><a href="#cb1-244" aria-hidden="true" tabindex="-1"></a>Most_Abun_Phylum <span class="ot">=</span> <span class="fu">as.character</span>(common_phylum_cor_df[,<span class="st">"Phylum"</span>])</span>
<span id="cb1-245"><a href="#cb1-245" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-246"><a href="#cb1-246" aria-hidden="true" tabindex="-1"></a>Most_Abun_Phylum_Plots <span class="ot">=</span> <span class="fu">lapply</span>(Most_Abun_Phylum, <span class="cf">function</span>(feature){</span>
<span id="cb1-247"><a href="#cb1-247" aria-hidden="true" tabindex="-1"></a>  feature_df <span class="ot">=</span> <span class="fu">data.frame</span>(<span class="at">rel_16s =</span> <span class="fu">as.vector</span>(physeq<span class="fl">.16</span>s.phylum.rel<span class="sc">@</span>otu_table[feature,]),</span>
<span id="cb1-248"><a href="#cb1-248" aria-hidden="true" tabindex="-1"></a>                          <span class="at">rel_wgs =</span> <span class="fu">as.vector</span>(physeq.wgs.phylum.rel<span class="sc">@</span>otu_table[feature,]))</span>
<span id="cb1-249"><a href="#cb1-249" aria-hidden="true" tabindex="-1"></a>  limit <span class="ot">=</span> <span class="fu">max</span>(feature_df)</span>
<span id="cb1-250"><a href="#cb1-250" aria-hidden="true" tabindex="-1"></a>  cor_value <span class="ot">=</span> <span class="fu">round</span>(common_phylum_cor_df[<span class="fu">which</span>(common_phylum_cor_df<span class="sc">$</span>Phylum <span class="sc">==</span> feature),<span class="st">"Corr"</span>],<span class="dv">2</span>)</span>
<span id="cb1-251"><a href="#cb1-251" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(feature_df, <span class="fu">aes</span>(<span class="at">x=</span>rel_16s, <span class="at">y=</span>rel_wgs))  <span class="sc">+</span> </span>
<span id="cb1-252"><a href="#cb1-252" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb1-253"><a href="#cb1-253" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_smooth</span>(<span class="at">method=</span>lm) <span class="sc">+</span></span>
<span id="cb1-254"><a href="#cb1-254" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_classic</span>()<span class="sc">+</span></span>
<span id="cb1-255"><a href="#cb1-255" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">title =</span> feature)<span class="sc">+</span></span>
<span id="cb1-256"><a href="#cb1-256" aria-hidden="true" tabindex="-1"></a>    <span class="fu">ylim</span>(<span class="dv">0</span>, limit)<span class="sc">+</span></span>
<span id="cb1-257"><a href="#cb1-257" aria-hidden="true" tabindex="-1"></a>    <span class="fu">xlim</span>(<span class="dv">0</span>, limit)<span class="sc">+</span></span>
<span id="cb1-258"><a href="#cb1-258" aria-hidden="true" tabindex="-1"></a>    <span class="fu">annotate</span>(<span class="st">"text"</span>,<span class="at">x =</span> <span class="dv">0</span>, <span class="at">y =</span> limit,<span class="at">label =</span> <span class="fu">paste0</span>(<span class="st">"rho = "</span>, cor_value),<span class="at">hjust =</span> <span class="sc">-</span><span class="fl">0.1</span>, <span class="at">vjust =</span> <span class="fl">1.1</span>,<span class="at">size =</span> <span class="dv">5</span>)</span>
<span id="cb1-259"><a href="#cb1-259" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb1-260"><a href="#cb1-260" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-261"><a href="#cb1-261" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb1-262"><a href="#cb1-262" aria-hidden="true" tabindex="-1"></a>Combined_Plots <span class="ot">&lt;-</span> <span class="fu">wrap_plots</span>(Most_Abun_Phylum_Plots, <span class="at">ncol =</span> <span class="dv">6</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="diversity-plot" class="level2">
<h2 class="anchored" data-anchor-id="diversity-plot">2. Diversity plot</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Alpha Diversity ----------------------------------------------------------------------------</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyr)</span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(vegan)</span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Rafefaction</span></span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a>x <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="fu">t</span>(<span class="fu">otu_table</span>(phylo_16s)))</span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="st">"Real_Plots/RF_16S.pdf"</span>, <span class="at">width =</span> <span class="dv">4</span>, <span class="at">height =</span> <span class="dv">3</span>) <span class="co"># Width and height are in inches for PDFs</span></span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a>rare_curve <span class="ot">&lt;-</span> <span class="fu">rarecurve</span>(x, <span class="at">step =</span> <span class="dv">2000</span>, <span class="at">sample =</span> <span class="dv">3000</span>, <span class="at">col =</span> <span class="st">"blue"</span>, <span class="at">cex =</span> <span class="fl">0.6</span>,<span class="at">label =</span> <span class="cn">FALSE</span>,<span class="at">xlab =</span><span class="st">""</span>)</span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a>x <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="fu">t</span>(<span class="fu">otu_table</span>(phylo_wgs)))</span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="st">"Real_Plots/RF_WGS.pdf"</span>, <span class="at">width =</span> <span class="dv">4</span>, <span class="at">height =</span> <span class="dv">3</span>) <span class="co"># Width and height are in inches for PDFs</span></span>
<span id="cb2-16"><a href="#cb2-16" aria-hidden="true" tabindex="-1"></a>rare_curve <span class="ot">&lt;-</span> <span class="fu">rarecurve</span>(x, <span class="at">step =</span> <span class="dv">2000</span>, <span class="at">sample =</span> <span class="dv">40000</span>, <span class="at">col =</span> <span class="st">"blue"</span>, <span class="at">cex =</span> <span class="fl">0.6</span>,<span class="at">label =</span> <span class="cn">FALSE</span>,<span class="at">xlab =</span><span class="st">""</span>)</span>
<span id="cb2-17"><a href="#cb2-17" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span>
<span id="cb2-18"><a href="#cb2-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-19"><a href="#cb2-19" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to calculate alpha diversity and reshape data</span></span>
<span id="cb2-20"><a href="#cb2-20" aria-hidden="true" tabindex="-1"></a>get_alpha_data <span class="ot">&lt;-</span> <span class="cf">function</span>(physeq, sample_size, sequencing_type) {</span>
<span id="cb2-21"><a href="#cb2-21" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Rarefy and get metadata</span></span>
<span id="cb2-22"><a href="#cb2-22" aria-hidden="true" tabindex="-1"></a>  physeq_rarefied <span class="ot">&lt;-</span> <span class="fu">rarefy_even_depth</span>(physeq, <span class="at">sample.size =</span> sample_size, <span class="at">rngseed =</span> <span class="dv">2024</span>)</span>
<span id="cb2-23"><a href="#cb2-23" aria-hidden="true" tabindex="-1"></a>  meta <span class="ot">&lt;-</span> <span class="fu">meta</span>(physeq_rarefied)</span>
<span id="cb2-24"><a href="#cb2-24" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb2-25"><a href="#cb2-25" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Calculate alpha diversity indices</span></span>
<span id="cb2-26"><a href="#cb2-26" aria-hidden="true" tabindex="-1"></a>  Alp_ind <span class="ot">&lt;-</span> <span class="fu">alpha</span>(physeq_rarefied, <span class="at">index =</span> <span class="fu">c</span>(<span class="st">"shannon"</span>, <span class="st">"inverse_simpson"</span>, <span class="st">"observed"</span>))</span>
<span id="cb2-27"><a href="#cb2-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(Alp_ind) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"observed"</span>, <span class="st">"shannon"</span>, <span class="st">"inverse_simpson"</span>)</span>
<span id="cb2-28"><a href="#cb2-28" aria-hidden="true" tabindex="-1"></a>  Alp_ind<span class="sc">$</span>diagnosis <span class="ot">&lt;-</span> <span class="fu">factor</span>(meta<span class="sc">$</span>diagnosis)</span>
<span id="cb2-29"><a href="#cb2-29" aria-hidden="true" tabindex="-1"></a>  Alp_ind<span class="sc">$</span>sequencing <span class="ot">&lt;-</span> sequencing_type</span>
<span id="cb2-30"><a href="#cb2-30" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb2-31"><a href="#cb2-31" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Convert to long format</span></span>
<span id="cb2-32"><a href="#cb2-32" aria-hidden="true" tabindex="-1"></a>  Alp_long <span class="ot">&lt;-</span> Alp_ind <span class="sc">%&gt;%</span></span>
<span id="cb2-33"><a href="#cb2-33" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">c</span>(<span class="st">"observed"</span>, <span class="st">"shannon"</span>, <span class="st">"inverse_simpson"</span>), </span>
<span id="cb2-34"><a href="#cb2-34" aria-hidden="true" tabindex="-1"></a>                 <span class="at">names_to =</span> <span class="st">"Index"</span>, <span class="at">values_to =</span> <span class="st">"Diversity"</span>)</span>
<span id="cb2-35"><a href="#cb2-35" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb2-36"><a href="#cb2-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(Alp_long)</span>
<span id="cb2-37"><a href="#cb2-37" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb2-38"><a href="#cb2-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-39"><a href="#cb2-39" aria-hidden="true" tabindex="-1"></a><span class="co"># Get alpha diversity data for 16S</span></span>
<span id="cb2-40"><a href="#cb2-40" aria-hidden="true" tabindex="-1"></a>alpha_16s <span class="ot">&lt;-</span> <span class="fu">get_alpha_data</span>(phylo_16s, <span class="at">sample_size =</span> <span class="dv">5000</span>, <span class="at">sequencing_type =</span> <span class="st">"16S"</span>)</span>
<span id="cb2-41"><a href="#cb2-41" aria-hidden="true" tabindex="-1"></a><span class="co"># Get alpha diversity data for WGS</span></span>
<span id="cb2-42"><a href="#cb2-42" aria-hidden="true" tabindex="-1"></a>alpha_wgs <span class="ot">&lt;-</span> <span class="fu">get_alpha_data</span>(phylo_wgs, <span class="at">sample_size =</span> <span class="dv">30000</span>, <span class="at">sequencing_type =</span> <span class="st">"WGS"</span>)</span>
<span id="cb2-43"><a href="#cb2-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-44"><a href="#cb2-44" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine both datasets</span></span>
<span id="cb2-45"><a href="#cb2-45" aria-hidden="true" tabindex="-1"></a>alpha_combined <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(alpha_16s, alpha_wgs)</span>
<span id="cb2-46"><a href="#cb2-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-47"><a href="#cb2-47" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to perform Wilcoxon test and add p-values</span></span>
<span id="cb2-48"><a href="#cb2-48" aria-hidden="true" tabindex="-1"></a>get_p_value <span class="ot">&lt;-</span> <span class="cf">function</span>(data) {</span>
<span id="cb2-49"><a href="#cb2-49" aria-hidden="true" tabindex="-1"></a>  test <span class="ot">&lt;-</span> <span class="fu">wilcox.test</span>(Diversity <span class="sc">~</span> diagnosis, <span class="at">data =</span> data, <span class="at">alternative =</span> <span class="st">"two.sided"</span>)</span>
<span id="cb2-50"><a href="#cb2-50" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(<span class="fu">round</span>(test<span class="sc">$</span>p.value, <span class="dv">4</span>))</span>
<span id="cb2-51"><a href="#cb2-51" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb2-52"><a href="#cb2-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-53"><a href="#cb2-53" aria-hidden="true" tabindex="-1"></a><span class="co"># Add p-values to the data frame</span></span>
<span id="cb2-54"><a href="#cb2-54" aria-hidden="true" tabindex="-1"></a>p_values <span class="ot">&lt;-</span> alpha_combined <span class="sc">%&gt;%</span></span>
<span id="cb2-55"><a href="#cb2-55" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sequencing, Index) <span class="sc">%&gt;%</span></span>
<span id="cb2-56"><a href="#cb2-56" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">p_value =</span> <span class="fu">get_p_value</span>(<span class="fu">cur_data</span>()))</span>
<span id="cb2-57"><a href="#cb2-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-58"><a href="#cb2-58" aria-hidden="true" tabindex="-1"></a><span class="co"># Merge p-values into the combined data</span></span>
<span id="cb2-59"><a href="#cb2-59" aria-hidden="true" tabindex="-1"></a>alpha_combined <span class="ot">&lt;-</span> <span class="fu">left_join</span>(alpha_combined, p_values, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"sequencing"</span>, <span class="st">"Index"</span>))</span>
<span id="cb2-60"><a href="#cb2-60" aria-hidden="true" tabindex="-1"></a>y <span class="ot">=</span> </span>
<span id="cb2-61"><a href="#cb2-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-62"><a href="#cb2-62" aria-hidden="true" tabindex="-1"></a>correlation_plot <span class="ot">&lt;-</span> <span class="cf">function</span>(x, y) {</span>
<span id="cb2-63"><a href="#cb2-63" aria-hidden="true" tabindex="-1"></a>  model <span class="ot">&lt;-</span> <span class="fu">lm</span>(y <span class="sc">~</span> x)</span>
<span id="cb2-64"><a href="#cb2-64" aria-hidden="true" tabindex="-1"></a>  sortx <span class="ot">&lt;-</span> <span class="fu">sort</span>(x)</span>
<span id="cb2-65"><a href="#cb2-65" aria-hidden="true" tabindex="-1"></a>  pred <span class="ot">&lt;-</span> <span class="fu">predict</span>(model, <span class="at">newdata =</span> <span class="fu">data.frame</span>(<span class="at">x =</span> x_sorted), <span class="at">interval =</span> <span class="st">"confidence"</span>)</span>
<span id="cb2-66"><a href="#cb2-66" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb2-67"><a href="#cb2-67" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Diversity Correlation heatscatter</span></span>
<span id="cb2-68"><a href="#cb2-68" aria-hidden="true" tabindex="-1"></a>  <span class="fu">heatscatter</span>(x, y,</span>
<span id="cb2-69"><a href="#cb2-69" aria-hidden="true" tabindex="-1"></a>              <span class="at">xlab =</span> <span class="st">"X"</span>,                      <span class="co"># X label</span></span>
<span id="cb2-70"><a href="#cb2-70" aria-hidden="true" tabindex="-1"></a>              <span class="at">ylab =</span> <span class="st">"Y"</span>,                      <span class="co"># Y label</span></span>
<span id="cb2-71"><a href="#cb2-71" aria-hidden="true" tabindex="-1"></a>              <span class="at">main =</span> <span class="st">"Correlation Plot"</span>,       <span class="co"># Main title</span></span>
<span id="cb2-72"><a href="#cb2-72" aria-hidden="true" tabindex="-1"></a>              <span class="at">color.palette =</span> <span class="fu">colorRampPalette</span>(<span class="fu">c</span>(<span class="st">"blue"</span>, <span class="st">"yellow"</span>, <span class="st">"red"</span>)),  <span class="co"># Color gradient</span></span>
<span id="cb2-73"><a href="#cb2-73" aria-hidden="true" tabindex="-1"></a>              <span class="at">cexplot =</span> <span class="fl">1.5</span>,                   <span class="co"># Point size</span></span>
<span id="cb2-74"><a href="#cb2-74" aria-hidden="true" tabindex="-1"></a>              <span class="at">pch =</span> <span class="dv">16</span>,                        <span class="co"># Point shape (circle)</span></span>
<span id="cb2-75"><a href="#cb2-75" aria-hidden="true" tabindex="-1"></a>              <span class="at">cex.lab =</span> <span class="fl">1.5</span>,                   <span class="co"># Axis label font size</span></span>
<span id="cb2-76"><a href="#cb2-76" aria-hidden="true" tabindex="-1"></a>              <span class="at">cex.axis =</span> <span class="fl">1.2</span>,                  <span class="co"># Axis tick font size</span></span>
<span id="cb2-77"><a href="#cb2-77" aria-hidden="true" tabindex="-1"></a>              <span class="at">cex.main =</span> <span class="fl">1.5</span>)                  <span class="co"># Title font size</span></span>
<span id="cb2-78"><a href="#cb2-78" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb2-79"><a href="#cb2-79" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Confidence band (shaded)</span></span>
<span id="cb2-80"><a href="#cb2-80" aria-hidden="true" tabindex="-1"></a>  <span class="fu">polygon</span>(<span class="fu">c</span>(x_sorted, <span class="fu">rev</span>(x_sorted)),              <span class="co"># X coordinates</span></span>
<span id="cb2-81"><a href="#cb2-81" aria-hidden="true" tabindex="-1"></a>          <span class="fu">c</span>(pred[, <span class="st">"lwr"</span>], <span class="fu">rev</span>(pred[, <span class="st">"upr"</span>])),    <span class="co"># Y coordinates</span></span>
<span id="cb2-82"><a href="#cb2-82" aria-hidden="true" tabindex="-1"></a>          <span class="at">col =</span> <span class="fu">rgb</span>(<span class="fl">0.7</span>, <span class="fl">0.7</span>, <span class="fl">0.7</span>, <span class="fl">0.4</span>),            <span class="co"># Color and transparency</span></span>
<span id="cb2-83"><a href="#cb2-83" aria-hidden="true" tabindex="-1"></a>          <span class="at">border =</span> <span class="cn">NA</span>)                             <span class="co"># Remove border</span></span>
<span id="cb2-84"><a href="#cb2-84" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb2-85"><a href="#cb2-85" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Regression line</span></span>
<span id="cb2-86"><a href="#cb2-86" aria-hidden="true" tabindex="-1"></a>  <span class="fu">lines</span>(x_sorted, pred[, <span class="st">"fit"</span>], <span class="at">col =</span> <span class="st">"green"</span>, <span class="at">lwd =</span> <span class="dv">3</span>)</span>
<span id="cb2-87"><a href="#cb2-87" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb2-88"><a href="#cb2-88" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-89"><a href="#cb2-89" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot using ggplot2 with facets</span></span>
<span id="cb2-90"><a href="#cb2-90" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggrain)</span>
<span id="cb2-91"><a href="#cb2-91" aria-hidden="true" tabindex="-1"></a>alpha_plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(alpha_combined, <span class="fu">aes</span>(<span class="at">x =</span> diagnosis, <span class="at">y =</span> Diversity, <span class="at">fill =</span> diagnosis)) <span class="sc">+</span></span>
<span id="cb2-92"><a href="#cb2-92" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_violin</span>(<span class="at">trim =</span> <span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb2-93"><a href="#cb2-93" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">width =</span> <span class="fl">0.2</span>, <span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="fl">0.9</span>), <span class="at">outlier.shape =</span> <span class="cn">NA</span>) <span class="sc">+</span></span>
<span id="cb2-94"><a href="#cb2-94" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb2-95"><a href="#cb2-95" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(Index <span class="sc">~</span> sequencing, <span class="at">scales =</span> <span class="st">"free_y"</span>) <span class="sc">+</span></span>
<span id="cb2-96"><a href="#cb2-96" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="cn">NULL</span>,<span class="at">y =</span> <span class="st">"Alpha Diversity"</span>) <span class="sc">+</span></span>
<span id="cb2-97"><a href="#cb2-97" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"bottom"</span>,</span>
<span id="cb2-98"><a href="#cb2-98" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.border =</span> <span class="fu">element_rect</span>(<span class="at">color =</span> <span class="st">"black"</span>, <span class="at">fill =</span> <span class="cn">NA</span>, <span class="at">linewidth =</span> <span class="fl">0.5</span>),</span>
<span id="cb2-99"><a href="#cb2-99" aria-hidden="true" tabindex="-1"></a>        <span class="at">strip.background =</span> <span class="fu">element_rect</span>(<span class="at">color =</span> <span class="st">"black"</span>, <span class="at">fill =</span> <span class="st">"gray90"</span>)) <span class="sc">+</span></span>
<span id="cb2-100"><a href="#cb2-100" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Annotate p-values in each facet</span></span>
<span id="cb2-101"><a href="#cb2-101" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(</span>
<span id="cb2-102"><a href="#cb2-102" aria-hidden="true" tabindex="-1"></a>    <span class="at">data =</span> p_values, </span>
<span id="cb2-103"><a href="#cb2-103" aria-hidden="true" tabindex="-1"></a>    <span class="fu">aes</span>(<span class="at">x =</span> <span class="fl">1.5</span>, <span class="at">y =</span> <span class="cn">Inf</span>, <span class="at">label =</span> <span class="fu">paste0</span>(<span class="st">"p = "</span>, <span class="fu">formatC</span>(p_value, <span class="at">format =</span> <span class="st">"f"</span>, <span class="at">digits =</span> <span class="dv">2</span>))),</span>
<span id="cb2-104"><a href="#cb2-104" aria-hidden="true" tabindex="-1"></a>    <span class="at">inherit.aes =</span> <span class="cn">FALSE</span>, </span>
<span id="cb2-105"><a href="#cb2-105" aria-hidden="true" tabindex="-1"></a>    <span class="at">vjust =</span> <span class="dv">2</span>, <span class="at">hjust =</span> <span class="fl">0.5</span>,<span class="at">size =</span> <span class="dv">4</span></span>
<span id="cb2-106"><a href="#cb2-106" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb2-107"><a href="#cb2-107" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-108"><a href="#cb2-108" aria-hidden="true" tabindex="-1"></a>alpha_plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(alpha_combined, <span class="fu">aes</span>(<span class="at">x =</span> diagnosis, <span class="at">y =</span> Diversity, <span class="at">fill =</span> diagnosis)) <span class="sc">+</span></span>
<span id="cb2-109"><a href="#cb2-109" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_rain</span>(<span class="at">violin.args =</span> <span class="fu">list</span>(<span class="at">alpha =</span> <span class="fl">0.7</span>),</span>
<span id="cb2-110"><a href="#cb2-110" aria-hidden="true" tabindex="-1"></a>            <span class="at">point.args =</span> <span class="fu">list</span>(<span class="at">size =</span> <span class="fl">0.3</span>, <span class="at">alpha =</span> <span class="dv">1</span>)) <span class="sc">+</span></span>
<span id="cb2-111"><a href="#cb2-111" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"uc"</span> <span class="ot">=</span> <span class="st">"#7AC3DF"</span>, <span class="st">"cd"</span> <span class="ot">=</span> <span class="st">"#EB7E60"</span>)) <span class="sc">+</span></span>
<span id="cb2-112"><a href="#cb2-112" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"uc"</span> <span class="ot">=</span> <span class="st">"#7AC3DF"</span>, <span class="st">"cd"</span> <span class="ot">=</span> <span class="st">"#EB7E60"</span>)) <span class="sc">+</span></span>
<span id="cb2-113"><a href="#cb2-113" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb2-114"><a href="#cb2-114" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(Index <span class="sc">~</span> sequencing, <span class="at">scales =</span> <span class="st">"free_y"</span>) <span class="sc">+</span></span>
<span id="cb2-115"><a href="#cb2-115" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="cn">NULL</span>, <span class="at">y =</span> <span class="cn">NULL</span>) <span class="sc">+</span></span>
<span id="cb2-116"><a href="#cb2-116" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb2-117"><a href="#cb2-117" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.position =</span> <span class="st">"none"</span>,</span>
<span id="cb2-118"><a href="#cb2-118" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.border =</span> <span class="fu">element_rect</span>(<span class="at">color =</span> <span class="st">"black"</span>, <span class="at">fill =</span> <span class="cn">NA</span>, <span class="at">linewidth =</span> <span class="dv">1</span>),</span>
<span id="cb2-119"><a href="#cb2-119" aria-hidden="true" tabindex="-1"></a>    <span class="at">strip.background =</span> <span class="fu">element_blank</span>(),     <span class="co"># Remove the grey boxes</span></span>
<span id="cb2-120"><a href="#cb2-120" aria-hidden="true" tabindex="-1"></a>    <span class="co"># strip.background = element_rect(color = "black", fill = "gray90"),</span></span>
<span id="cb2-121"><a href="#cb2-121" aria-hidden="true" tabindex="-1"></a>    <span class="co"># axis.text.y = element_blank(),     # Remove y-axis text</span></span>
<span id="cb2-122"><a href="#cb2-122" aria-hidden="true" tabindex="-1"></a>    <span class="co"># axis.ticks.y = element_blank(),    # Remove y-axis ticks</span></span>
<span id="cb2-123"><a href="#cb2-123" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.line =</span> <span class="fu">element_blank</span>()</span>
<span id="cb2-124"><a href="#cb2-124" aria-hidden="true" tabindex="-1"></a>  ) </span>
<span id="cb2-125"><a href="#cb2-125" aria-hidden="true" tabindex="-1"></a>alpha_plot</span>
<span id="cb2-126"><a href="#cb2-126" aria-hidden="true" tabindex="-1"></a><span class="co"># Save the plot</span></span>
<span id="cb2-127"><a href="#cb2-127" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"Real_Plots/Alpha_Diversity_combined.pdf"</span>, alpha_plot, <span class="at">width =</span> <span class="dv">5</span>, <span class="at">height =</span> <span class="dv">5</span>)</span>
<span id="cb2-128"><a href="#cb2-128" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-129"><a href="#cb2-129" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-130"><a href="#cb2-130" aria-hidden="true" tabindex="-1"></a><span class="do">## Alpha Diversity Correlation----------------------------------------------------------------------------</span></span>
<span id="cb2-131"><a href="#cb2-131" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(energy)</span>
<span id="cb2-132"><a href="#cb2-132" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(XICOR)</span>
<span id="cb2-133"><a href="#cb2-133" aria-hidden="true" tabindex="-1"></a><span class="fu">sample_names</span>(phylo_16s) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"-16s$"</span>, <span class="st">""</span>, <span class="fu">sample_names</span>(phylo_16s))</span>
<span id="cb2-134"><a href="#cb2-134" aria-hidden="true" tabindex="-1"></a>physeq<span class="fl">.16</span>s <span class="ot">=</span> <span class="fu">rarefy_even_depth</span>(phylo_16s,<span class="at">sample.size =</span> <span class="dv">5000</span>,<span class="at">rngseed =</span> <span class="dv">2024</span>)</span>
<span id="cb2-135"><a href="#cb2-135" aria-hidden="true" tabindex="-1"></a>physeq.wgs <span class="ot">=</span> <span class="fu">rarefy_even_depth</span>(phylo_wgs,<span class="at">sample.size =</span> <span class="dv">30000</span>,<span class="at">rngseed =</span> <span class="dv">2024</span>)</span>
<span id="cb2-136"><a href="#cb2-136" aria-hidden="true" tabindex="-1"></a>common_samples <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">sample_names</span>(physeq<span class="fl">.16</span>s), <span class="fu">sample_names</span>(physeq.wgs))</span>
<span id="cb2-137"><a href="#cb2-137" aria-hidden="true" tabindex="-1"></a>physeq<span class="fl">.16</span>s <span class="ot">&lt;-</span> <span class="fu">prune_samples</span>(common_samples, physeq<span class="fl">.16</span>s)</span>
<span id="cb2-138"><a href="#cb2-138" aria-hidden="true" tabindex="-1"></a>physeq.wgs <span class="ot">&lt;-</span> <span class="fu">prune_samples</span>(common_samples, physeq.wgs)</span>
<span id="cb2-139"><a href="#cb2-139" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-140"><a href="#cb2-140" aria-hidden="true" tabindex="-1"></a>Alp_ind_16s <span class="ot">=</span> <span class="fu">alpha</span>(physeq<span class="fl">.16</span>s,<span class="at">index =</span> <span class="fu">c</span>(<span class="st">"shannon"</span>,<span class="st">"inverse_simpson"</span>,<span class="st">"observed"</span>))</span>
<span id="cb2-141"><a href="#cb2-141" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(Alp_ind_16s) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"observed"</span>,<span class="st">"shannon"</span>,<span class="st">"inverse_simpson"</span>)</span>
<span id="cb2-142"><a href="#cb2-142" aria-hidden="true" tabindex="-1"></a>Alp_ind_wgs <span class="ot">=</span> <span class="fu">alpha</span>(physeq.wgs,<span class="at">index =</span> <span class="fu">c</span>(<span class="st">"shannon"</span>,<span class="st">"inverse_simpson"</span>,<span class="st">"observed"</span>))</span>
<span id="cb2-143"><a href="#cb2-143" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(Alp_ind_wgs) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"observed"</span>,<span class="st">"shannon"</span>,<span class="st">"inverse_simpson"</span>)</span>
<span id="cb2-144"><a href="#cb2-144" aria-hidden="true" tabindex="-1"></a>Alp_ind_wgs <span class="ot">&lt;-</span> Alp_ind_wgs[<span class="fu">match</span>(<span class="fu">rownames</span>(Alp_ind_16s), <span class="fu">rownames</span>(Alp_ind_wgs)), ]</span>
<span id="cb2-145"><a href="#cb2-145" aria-hidden="true" tabindex="-1"></a><span class="fu">all</span>(<span class="fu">rownames</span>(Alp_ind_16s) <span class="sc">==</span> <span class="fu">rownames</span>(Alp_ind_wgs)) <span class="do">## make sure order is matched</span></span>
<span id="cb2-146"><a href="#cb2-146" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-147"><a href="#cb2-147" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb2-148"><a href="#cb2-148" aria-hidden="true" tabindex="-1"></a><span class="co"># Step 1: Combine and reshape data</span></span>
<span id="cb2-149"><a href="#cb2-149" aria-hidden="true" tabindex="-1"></a>index_list <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"observed"</span>, <span class="st">"shannon"</span>, <span class="st">"inverse_simpson"</span>)</span>
<span id="cb2-150"><a href="#cb2-150" aria-hidden="true" tabindex="-1"></a>long_df <span class="ot">&lt;-</span> <span class="fu">do.call</span>(rbind, <span class="fu">lapply</span>(index_list, <span class="cf">function</span>(index) {</span>
<span id="cb2-151"><a href="#cb2-151" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data.frame</span>(</span>
<span id="cb2-152"><a href="#cb2-152" aria-hidden="true" tabindex="-1"></a>    <span class="at">Index =</span> index,</span>
<span id="cb2-153"><a href="#cb2-153" aria-hidden="true" tabindex="-1"></a>    <span class="at">Alp_16s =</span> Alp_ind_16s[[index]],</span>
<span id="cb2-154"><a href="#cb2-154" aria-hidden="true" tabindex="-1"></a>    <span class="at">Alp_wgs =</span> Alp_ind_wgs[[index]]</span>
<span id="cb2-155"><a href="#cb2-155" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb2-156"><a href="#cb2-156" aria-hidden="true" tabindex="-1"></a>}))</span>
<span id="cb2-157"><a href="#cb2-157" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-158"><a href="#cb2-158" aria-hidden="true" tabindex="-1"></a>long_df_observed <span class="ot">&lt;-</span> long_df <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Index<span class="sc">==</span><span class="st">"observed"</span>)</span>
<span id="cb2-159"><a href="#cb2-159" aria-hidden="true" tabindex="-1"></a>x <span class="ot">=</span> long_df_observed<span class="sc">$</span>Alp_16s</span>
<span id="cb2-160"><a href="#cb2-160" aria-hidden="true" tabindex="-1"></a>y <span class="ot">=</span> long_df_observed<span class="sc">$</span>Alp_wgs</span>
<span id="cb2-161"><a href="#cb2-161" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">correlation_plot</span>(x,y)</span>
<span id="cb2-162"><a href="#cb2-162" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-163"><a href="#cb2-163" aria-hidden="true" tabindex="-1"></a>plot_correlations_by_index <span class="ot">&lt;-</span> <span class="cf">function</span>(df, <span class="at">index_col =</span> <span class="st">"Index"</span>, <span class="at">x_col =</span> <span class="st">"Alp_16s"</span>, <span class="at">y_col =</span> <span class="st">"Alp_wgs"</span>, <span class="at">outdir =</span> <span class="st">"."</span>) {</span>
<span id="cb2-164"><a href="#cb2-164" aria-hidden="true" tabindex="-1"></a>  unique_indices <span class="ot">&lt;-</span> <span class="fu">unique</span>(df[[index_col]])</span>
<span id="cb2-165"><a href="#cb2-165" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb2-166"><a href="#cb2-166" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (index <span class="cf">in</span> unique_indices) {</span>
<span id="cb2-167"><a href="#cb2-167" aria-hidden="true" tabindex="-1"></a>    sub_df <span class="ot">&lt;-</span> df <span class="sc">%&gt;%</span> <span class="fu">filter</span>(.data[[index_col]] <span class="sc">==</span> index)</span>
<span id="cb2-168"><a href="#cb2-168" aria-hidden="true" tabindex="-1"></a>    x <span class="ot">&lt;-</span> sub_df[[x_col]]</span>
<span id="cb2-169"><a href="#cb2-169" aria-hidden="true" tabindex="-1"></a>    y <span class="ot">&lt;-</span> sub_df[[y_col]]</span>
<span id="cb2-170"><a href="#cb2-170" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb2-171"><a href="#cb2-171" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Create a safe filename</span></span>
<span id="cb2-172"><a href="#cb2-172" aria-hidden="true" tabindex="-1"></a>    filename <span class="ot">&lt;-</span> <span class="fu">file.path</span>(outdir, <span class="fu">paste0</span>(<span class="st">"correlation_plot_"</span>, <span class="fu">gsub</span>(<span class="st">"[^A-Za-z0-9]"</span>, <span class="st">"_"</span>, index), <span class="st">".pdf"</span>))</span>
<span id="cb2-173"><a href="#cb2-173" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb2-174"><a href="#cb2-174" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Save to PDF</span></span>
<span id="cb2-175"><a href="#cb2-175" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pdf</span>(filename, <span class="at">width =</span> <span class="dv">10</span>, <span class="at">height =</span> <span class="dv">6</span>)</span>
<span id="cb2-176"><a href="#cb2-176" aria-hidden="true" tabindex="-1"></a>    <span class="fu">correlation_plot</span>(x, y)</span>
<span id="cb2-177"><a href="#cb2-177" aria-hidden="true" tabindex="-1"></a>    <span class="co"># title(main = paste("Correlation Plot -", index))</span></span>
<span id="cb2-178"><a href="#cb2-178" aria-hidden="true" tabindex="-1"></a>    <span class="fu">dev.off</span>()</span>
<span id="cb2-179"><a href="#cb2-179" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb2-180"><a href="#cb2-180" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb2-181"><a href="#cb2-181" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-182"><a href="#cb2-182" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb2-183"><a href="#cb2-183" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb2-184"><a href="#cb2-184" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb2-185"><a href="#cb2-185" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-186"><a href="#cb2-186" aria-hidden="true" tabindex="-1"></a>plot_correlations_by_index <span class="ot">&lt;-</span> <span class="cf">function</span>(df, <span class="at">index_col =</span> <span class="st">"Index"</span>, <span class="at">x_col =</span> <span class="st">"Alp_16s"</span>, <span class="at">y_col =</span> <span class="st">"Alp_wgs"</span>, <span class="at">outdir =</span> <span class="st">"."</span>) {</span>
<span id="cb2-187"><a href="#cb2-187" aria-hidden="true" tabindex="-1"></a>  unique_indices <span class="ot">&lt;-</span> <span class="fu">unique</span>(df[[index_col]])</span>
<span id="cb2-188"><a href="#cb2-188" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb2-189"><a href="#cb2-189" aria-hidden="true" tabindex="-1"></a>  plot_list <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb2-190"><a href="#cb2-190" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb2-191"><a href="#cb2-191" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (index <span class="cf">in</span> unique_indices) {</span>
<span id="cb2-192"><a href="#cb2-192" aria-hidden="true" tabindex="-1"></a>    sub_df <span class="ot">&lt;-</span> df <span class="sc">%&gt;%</span> <span class="fu">filter</span>(.data[[index_col]] <span class="sc">==</span> index)</span>
<span id="cb2-193"><a href="#cb2-193" aria-hidden="true" tabindex="-1"></a>    x <span class="ot">&lt;-</span> sub_df[[x_col]]</span>
<span id="cb2-194"><a href="#cb2-194" aria-hidden="true" tabindex="-1"></a>    y <span class="ot">&lt;-</span> sub_df[[y_col]]</span>
<span id="cb2-195"><a href="#cb2-195" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb2-196"><a href="#cb2-196" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Generate plot with title</span></span>
<span id="cb2-197"><a href="#cb2-197" aria-hidden="true" tabindex="-1"></a>    p <span class="ot">&lt;-</span> <span class="fu">correlation_plot</span>(x, y)</span>
<span id="cb2-198"><a href="#cb2-198" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb2-199"><a href="#cb2-199" aria-hidden="true" tabindex="-1"></a>    plot_list[[index]] <span class="ot">&lt;-</span> p</span>
<span id="cb2-200"><a href="#cb2-200" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb2-201"><a href="#cb2-201" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb2-202"><a href="#cb2-202" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Combine plots vertically (ncol = 1)</span></span>
<span id="cb2-203"><a href="#cb2-203" aria-hidden="true" tabindex="-1"></a>  combined_plot <span class="ot">&lt;-</span> <span class="fu">wrap_plots</span>(plot_list, <span class="at">ncol =</span> <span class="dv">1</span>)</span>
<span id="cb2-204"><a href="#cb2-204" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb2-205"><a href="#cb2-205" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Save combined plot</span></span>
<span id="cb2-206"><a href="#cb2-206" aria-hidden="true" tabindex="-1"></a>  filename <span class="ot">&lt;-</span> <span class="fu">file.path</span>(outdir, <span class="st">"alpha_correlation_plots_combined.pdf"</span>)</span>
<span id="cb2-207"><a href="#cb2-207" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggsave</span>(filename, <span class="at">plot =</span> combined_plot, <span class="at">width =</span> <span class="dv">10</span>, <span class="at">height =</span> <span class="dv">6</span> <span class="sc">*</span> <span class="fu">length</span>(plot_list))</span>
<span id="cb2-208"><a href="#cb2-208" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb2-209"><a href="#cb2-209" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-210"><a href="#cb2-210" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-211"><a href="#cb2-211" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-212"><a href="#cb2-212" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_correlations_by_index</span>(long_df)</span>
<span id="cb2-213"><a href="#cb2-213" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-214"><a href="#cb2-214" aria-hidden="true" tabindex="-1"></a>correlation_plot <span class="ot">&lt;-</span> <span class="cf">function</span>(x, y) {</span>
<span id="cb2-215"><a href="#cb2-215" aria-hidden="true" tabindex="-1"></a>  <span class="fu">par</span>(<span class="at">bg=</span><span class="st">"white"</span>)</span>
<span id="cb2-216"><a href="#cb2-216" aria-hidden="true" tabindex="-1"></a>  model <span class="ot">&lt;-</span> <span class="fu">lm</span>(y <span class="sc">~</span> x)</span>
<span id="cb2-217"><a href="#cb2-217" aria-hidden="true" tabindex="-1"></a>  x_sorted <span class="ot">&lt;-</span> <span class="fu">sort</span>(x)</span>
<span id="cb2-218"><a href="#cb2-218" aria-hidden="true" tabindex="-1"></a>  pred <span class="ot">&lt;-</span> <span class="fu">predict</span>(model, <span class="at">newdata =</span> <span class="fu">data.frame</span>(<span class="at">x =</span> x_sorted), <span class="at">interval =</span> <span class="st">"confidence"</span>)</span>
<span id="cb2-219"><a href="#cb2-219" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb2-220"><a href="#cb2-220" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Diversity Correlation heatscatter</span></span>
<span id="cb2-221"><a href="#cb2-221" aria-hidden="true" tabindex="-1"></a>  <span class="fu">heatscatter</span>(x, y,</span>
<span id="cb2-222"><a href="#cb2-222" aria-hidden="true" tabindex="-1"></a>              <span class="at">xlab =</span> <span class="st">" "</span>,                      <span class="co"># X label</span></span>
<span id="cb2-223"><a href="#cb2-223" aria-hidden="true" tabindex="-1"></a>              <span class="at">ylab =</span> <span class="st">" "</span>,                      <span class="co"># Y label</span></span>
<span id="cb2-224"><a href="#cb2-224" aria-hidden="true" tabindex="-1"></a>              <span class="at">main =</span> <span class="st">" "</span>,       <span class="co"># Main title</span></span>
<span id="cb2-225"><a href="#cb2-225" aria-hidden="true" tabindex="-1"></a>              <span class="at">color.palette =</span> <span class="fu">colorRampPalette</span>(<span class="fu">c</span>(<span class="st">"blue"</span>, <span class="st">"yellow"</span>, <span class="st">"red"</span>)),  <span class="co"># Color gradient</span></span>
<span id="cb2-226"><a href="#cb2-226" aria-hidden="true" tabindex="-1"></a>              <span class="at">cexplot =</span> <span class="fl">1.2</span>,                   <span class="co"># Point size</span></span>
<span id="cb2-227"><a href="#cb2-227" aria-hidden="true" tabindex="-1"></a>              <span class="at">pch =</span> <span class="dv">16</span>,                        <span class="co"># Point shape (circle)</span></span>
<span id="cb2-228"><a href="#cb2-228" aria-hidden="true" tabindex="-1"></a>              <span class="at">cex.lab =</span> <span class="fl">1.5</span>,                   <span class="co"># Axis label font size</span></span>
<span id="cb2-229"><a href="#cb2-229" aria-hidden="true" tabindex="-1"></a>              <span class="at">cex.axis =</span> <span class="dv">2</span>,                  <span class="co"># Axis tick font size</span></span>
<span id="cb2-230"><a href="#cb2-230" aria-hidden="true" tabindex="-1"></a>              <span class="at">cex.main =</span> <span class="fl">1.5</span>)                  <span class="co"># Title font size</span></span>
<span id="cb2-231"><a href="#cb2-231" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb2-232"><a href="#cb2-232" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb2-233"><a href="#cb2-233" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Confidence band (shaded)</span></span>
<span id="cb2-234"><a href="#cb2-234" aria-hidden="true" tabindex="-1"></a>  <span class="fu">polygon</span>(<span class="fu">c</span>(x_sorted, <span class="fu">rev</span>(x_sorted)),              <span class="co"># X coordinates</span></span>
<span id="cb2-235"><a href="#cb2-235" aria-hidden="true" tabindex="-1"></a>          <span class="fu">c</span>(pred[, <span class="st">"lwr"</span>], <span class="fu">rev</span>(pred[, <span class="st">"upr"</span>])),    <span class="co"># Y coordinates</span></span>
<span id="cb2-236"><a href="#cb2-236" aria-hidden="true" tabindex="-1"></a>          <span class="at">col =</span> <span class="fu">rgb</span>(<span class="fl">0.7</span>, <span class="fl">0.7</span>, <span class="fl">0.7</span>, <span class="fl">0.4</span>),            <span class="co"># Color and transparency</span></span>
<span id="cb2-237"><a href="#cb2-237" aria-hidden="true" tabindex="-1"></a>          <span class="at">border =</span> <span class="cn">NA</span>)                             <span class="co"># Remove border</span></span>
<span id="cb2-238"><a href="#cb2-238" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb2-239"><a href="#cb2-239" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Regression line</span></span>
<span id="cb2-240"><a href="#cb2-240" aria-hidden="true" tabindex="-1"></a>  <span class="fu">lines</span>(x_sorted, pred[, <span class="st">"fit"</span>], <span class="at">col =</span> <span class="st">"green"</span>, <span class="at">lwd =</span> <span class="dv">3</span>)</span>
<span id="cb2-241"><a href="#cb2-241" aria-hidden="true" tabindex="-1"></a>  <span class="fu">box</span>(<span class="at">lwd=</span><span class="fl">1.5</span>)</span>
<span id="cb2-242"><a href="#cb2-242" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb2-243"><a href="#cb2-243" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-244"><a href="#cb2-244" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-245"><a href="#cb2-245" aria-hidden="true" tabindex="-1"></a><span class="co"># Step 2: Plot with facet_wrap</span></span>
<span id="cb2-246"><a href="#cb2-246" aria-hidden="true" tabindex="-1"></a>Combined_Plots <span class="ot">=</span> <span class="fu">ggplot</span>(long_df, <span class="fu">aes</span>(<span class="at">x =</span> Alp_16s, <span class="at">y =</span> Alp_wgs)) <span class="sc">+</span></span>
<span id="cb2-247"><a href="#cb2-247" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb2-248"><a href="#cb2-248" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">"lm"</span>, <span class="at">se =</span> <span class="cn">FALSE</span>,<span class="at">color =</span> <span class="st">"#0098B9"</span>) <span class="sc">+</span></span>
<span id="cb2-249"><a href="#cb2-249" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>Index, <span class="at">scales =</span> <span class="st">"free"</span>, <span class="at">ncol =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb2-250"><a href="#cb2-250" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb2-251"><a href="#cb2-251" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"16S"</span>, <span class="at">y =</span> <span class="st">"Shotgun"</span>, <span class="at">title =</span> <span class="cn">NULL</span>)</span>
<span id="cb2-252"><a href="#cb2-252" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-253"><a href="#cb2-253" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-254"><a href="#cb2-254" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"Real_Plots/Alpha_Diversity_Cor_1.pdf"</span>,Combined_Plots ,<span class="at">width =</span> <span class="dv">3</span>,<span class="at">height =</span> <span class="dv">4</span>)</span>
<span id="cb2-255"><a href="#cb2-255" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-256"><a href="#cb2-256" aria-hidden="true" tabindex="-1"></a><span class="do">## Beta Diversity-16s -----------------------------------------------------------------------</span></span>
<span id="cb2-257"><a href="#cb2-257" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(vegan)</span>
<span id="cb2-258"><a href="#cb2-258" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-259"><a href="#cb2-259" aria-hidden="true" tabindex="-1"></a>physeq <span class="ot">=</span> <span class="fu">transform_sample_counts</span>(phylo_16s, <span class="cf">function</span>(x) x <span class="sc">/</span> <span class="fu">sum</span>(x))</span>
<span id="cb2-260"><a href="#cb2-260" aria-hidden="true" tabindex="-1"></a>meta <span class="ot">=</span> <span class="fu">meta</span>(physeq)</span>
<span id="cb2-261"><a href="#cb2-261" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-262"><a href="#cb2-262" aria-hidden="true" tabindex="-1"></a>Beta_plots1 <span class="ot">=</span> <span class="fu">lapply</span>(<span class="fu">c</span>(<span class="st">"bray"</span>,<span class="st">"unifrac"</span>,<span class="st">"wunifrac"</span>), <span class="cf">function</span>(metric){</span>
<span id="cb2-263"><a href="#cb2-263" aria-hidden="true" tabindex="-1"></a>  permanova_result <span class="ot">&lt;-</span> <span class="fu">adonis2</span>(phyloseq<span class="sc">::</span><span class="fu">distance</span>(physeq, <span class="at">method=</span>metric) <span class="sc">~</span> diagnosis, </span>
<span id="cb2-264"><a href="#cb2-264" aria-hidden="true" tabindex="-1"></a>                             <span class="at">data=</span><span class="fu">data.frame</span>(physeq<span class="sc">@</span>sam_data))</span>
<span id="cb2-265"><a href="#cb2-265" aria-hidden="true" tabindex="-1"></a>  p_val <span class="ot">&lt;-</span> permanova_result<span class="sc">$</span><span class="st">"Pr(&gt;F)"</span>[<span class="dv">1</span>]</span>
<span id="cb2-266"><a href="#cb2-266" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(p_val)</span>
<span id="cb2-267"><a href="#cb2-267" aria-hidden="true" tabindex="-1"></a>  ord <span class="ot">=</span> <span class="fu">ordinate</span>(physeq, <span class="at">method=</span><span class="st">"PCoA"</span>, <span class="at">distance =</span> metric)</span>
<span id="cb2-268"><a href="#cb2-268" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb2-269"><a href="#cb2-269" aria-hidden="true" tabindex="-1"></a>  eig_vals <span class="ot">&lt;-</span> ord<span class="sc">$</span>values<span class="sc">$</span>Relative_eig <span class="sc">*</span> <span class="dv">100</span></span>
<span id="cb2-270"><a href="#cb2-270" aria-hidden="true" tabindex="-1"></a>  xlab <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"PCoA1 ["</span>, <span class="fu">round</span>(eig_vals[<span class="dv">1</span>], <span class="dv">1</span>), <span class="st">"%]"</span>)</span>
<span id="cb2-271"><a href="#cb2-271" aria-hidden="true" tabindex="-1"></a>  ylab <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"PCoA2 ["</span>, <span class="fu">round</span>(eig_vals[<span class="dv">2</span>], <span class="dv">1</span>), <span class="st">"%]"</span>)</span>
<span id="cb2-272"><a href="#cb2-272" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb2-273"><a href="#cb2-273" aria-hidden="true" tabindex="-1"></a>  ord_df <span class="ot">&lt;-</span> <span class="fu">plot_ordination</span>(physeq, ord, <span class="at">color =</span> <span class="st">"diagnosis"</span>, <span class="at">justDF =</span> <span class="cn">TRUE</span>)</span>
<span id="cb2-274"><a href="#cb2-274" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb2-275"><a href="#cb2-275" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(ord_df, <span class="fu">aes</span>(<span class="at">x =</span> Axis<span class="fl">.1</span>, <span class="at">y =</span> Axis<span class="fl">.2</span>, <span class="at">color =</span> diagnosis, <span class="at">fill =</span> diagnosis)) <span class="sc">+</span></span>
<span id="cb2-276"><a href="#cb2-276" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="fl">0.5</span>, <span class="at">alpha =</span> <span class="fl">0.7</span>) <span class="sc">+</span>  <span class="co"># control point size here</span></span>
<span id="cb2-277"><a href="#cb2-277" aria-hidden="true" tabindex="-1"></a>    <span class="fu">stat_ellipse</span>(<span class="fu">aes</span>(<span class="at">group =</span> diagnosis)) <span class="sc">+</span></span>
<span id="cb2-278"><a href="#cb2-278" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"uc"</span> <span class="ot">=</span> <span class="st">"#7AC3DF"</span>, <span class="st">"cd"</span> <span class="ot">=</span> <span class="st">"#EB7E60"</span>)) <span class="sc">+</span></span>
<span id="cb2-279"><a href="#cb2-279" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"uc"</span> <span class="ot">=</span> <span class="st">"#7AC3DF"</span>, <span class="st">"cd"</span> <span class="ot">=</span> <span class="st">"#EB7E60"</span>)) <span class="sc">+</span></span>
<span id="cb2-280"><a href="#cb2-280" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb2-281"><a href="#cb2-281" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">title =</span> <span class="cn">NULL</span>,<span class="at">x =</span> xlab, <span class="at">y =</span> ylab) <span class="sc">+</span></span>
<span id="cb2-282"><a href="#cb2-282" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(</span>
<span id="cb2-283"><a href="#cb2-283" aria-hidden="true" tabindex="-1"></a>      <span class="at">axis.ticks =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb2-284"><a href="#cb2-284" aria-hidden="true" tabindex="-1"></a>      <span class="at">axis.text =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb2-285"><a href="#cb2-285" aria-hidden="true" tabindex="-1"></a>      <span class="at">legend.position =</span> <span class="st">"none"</span></span>
<span id="cb2-286"><a href="#cb2-286" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb2-287"><a href="#cb2-287" aria-hidden="true" tabindex="-1"></a>    <span class="co"># +annotate("text", x = Inf, y = Inf, label = paste("p =", round(p_val,2)), </span></span>
<span id="cb2-288"><a href="#cb2-288" aria-hidden="true" tabindex="-1"></a>    <span class="co">#         hjust = 2, vjust = 1.1)</span></span>
<span id="cb2-289"><a href="#cb2-289" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb2-290"><a href="#cb2-290" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb2-291"><a href="#cb2-291" aria-hidden="true" tabindex="-1"></a><span class="do">## Beta Diversity-WGS -----------------------------------------------------------------------</span></span>
<span id="cb2-292"><a href="#cb2-292" aria-hidden="true" tabindex="-1"></a>physeq <span class="ot">=</span> <span class="fu">transform_sample_counts</span>(phylo_wgs, <span class="cf">function</span>(x) x <span class="sc">/</span> <span class="fu">sum</span>(x))</span>
<span id="cb2-293"><a href="#cb2-293" aria-hidden="true" tabindex="-1"></a>meta <span class="ot">=</span> <span class="fu">meta</span>(physeq)</span>
<span id="cb2-294"><a href="#cb2-294" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-295"><a href="#cb2-295" aria-hidden="true" tabindex="-1"></a>Beta_plots2 <span class="ot">=</span> <span class="fu">lapply</span>(<span class="fu">c</span>(<span class="st">"bray"</span>,<span class="st">"unifrac"</span>,<span class="st">"wunifrac"</span>), <span class="cf">function</span>(metric){</span>
<span id="cb2-296"><a href="#cb2-296" aria-hidden="true" tabindex="-1"></a>  permanova_result <span class="ot">&lt;-</span> <span class="fu">adonis2</span>(phyloseq<span class="sc">::</span><span class="fu">distance</span>(physeq, <span class="at">method=</span>metric) <span class="sc">~</span> diagnosis, </span>
<span id="cb2-297"><a href="#cb2-297" aria-hidden="true" tabindex="-1"></a>                             <span class="at">data=</span><span class="fu">data.frame</span>(physeq<span class="sc">@</span>sam_data))</span>
<span id="cb2-298"><a href="#cb2-298" aria-hidden="true" tabindex="-1"></a>  p_val <span class="ot">&lt;-</span> permanova_result<span class="sc">$</span><span class="st">"Pr(&gt;F)"</span>[<span class="dv">1</span>]</span>
<span id="cb2-299"><a href="#cb2-299" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(p_val)</span>
<span id="cb2-300"><a href="#cb2-300" aria-hidden="true" tabindex="-1"></a>  ord <span class="ot">=</span> <span class="fu">ordinate</span>(physeq, <span class="at">method=</span><span class="st">"PCoA"</span>, <span class="at">distance =</span> metric)</span>
<span id="cb2-301"><a href="#cb2-301" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb2-302"><a href="#cb2-302" aria-hidden="true" tabindex="-1"></a>  eig_vals <span class="ot">&lt;-</span> ord<span class="sc">$</span>values<span class="sc">$</span>Relative_eig <span class="sc">*</span> <span class="dv">100</span></span>
<span id="cb2-303"><a href="#cb2-303" aria-hidden="true" tabindex="-1"></a>  xlab <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"PCoA1 ["</span>, <span class="fu">round</span>(eig_vals[<span class="dv">1</span>], <span class="dv">1</span>), <span class="st">"%]"</span>)</span>
<span id="cb2-304"><a href="#cb2-304" aria-hidden="true" tabindex="-1"></a>  ylab <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"PCoA2 ["</span>, <span class="fu">round</span>(eig_vals[<span class="dv">2</span>], <span class="dv">1</span>), <span class="st">"%]"</span>)</span>
<span id="cb2-305"><a href="#cb2-305" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb2-306"><a href="#cb2-306" aria-hidden="true" tabindex="-1"></a>  ord_df <span class="ot">&lt;-</span> <span class="fu">plot_ordination</span>(physeq, ord, <span class="at">color =</span> <span class="st">"diagnosis"</span>, <span class="at">justDF =</span> <span class="cn">TRUE</span>)</span>
<span id="cb2-307"><a href="#cb2-307" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb2-308"><a href="#cb2-308" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(ord_df, <span class="fu">aes</span>(<span class="at">x =</span> Axis<span class="fl">.1</span>, <span class="at">y =</span> Axis<span class="fl">.2</span>, <span class="at">color =</span> diagnosis, <span class="at">fill =</span> diagnosis)) <span class="sc">+</span></span>
<span id="cb2-309"><a href="#cb2-309" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="fl">0.5</span>, <span class="at">alpha =</span> <span class="fl">0.7</span>) <span class="sc">+</span>  <span class="co"># control point size here</span></span>
<span id="cb2-310"><a href="#cb2-310" aria-hidden="true" tabindex="-1"></a>    <span class="fu">stat_ellipse</span>(<span class="fu">aes</span>(<span class="at">group =</span> diagnosis)) <span class="sc">+</span></span>
<span id="cb2-311"><a href="#cb2-311" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"uc"</span> <span class="ot">=</span> <span class="st">"#7AC3DF"</span>, <span class="st">"cd"</span> <span class="ot">=</span> <span class="st">"#EB7E60"</span>)) <span class="sc">+</span></span>
<span id="cb2-312"><a href="#cb2-312" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"uc"</span> <span class="ot">=</span> <span class="st">"#7AC3DF"</span>, <span class="st">"cd"</span> <span class="ot">=</span> <span class="st">"#EB7E60"</span>)) <span class="sc">+</span></span>
<span id="cb2-313"><a href="#cb2-313" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb2-314"><a href="#cb2-314" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">title =</span> <span class="cn">NULL</span>,<span class="at">x =</span> xlab, <span class="at">y =</span> ylab) <span class="sc">+</span></span>
<span id="cb2-315"><a href="#cb2-315" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(</span>
<span id="cb2-316"><a href="#cb2-316" aria-hidden="true" tabindex="-1"></a>      <span class="at">axis.ticks =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb2-317"><a href="#cb2-317" aria-hidden="true" tabindex="-1"></a>      <span class="at">axis.text =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb2-318"><a href="#cb2-318" aria-hidden="true" tabindex="-1"></a>      <span class="at">legend.position =</span> <span class="st">"none"</span></span>
<span id="cb2-319"><a href="#cb2-319" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb2-320"><a href="#cb2-320" aria-hidden="true" tabindex="-1"></a>  <span class="co"># +annotate("text", x = Inf, y = Inf, label = paste("p =", round(p_val,2)), </span></span>
<span id="cb2-321"><a href="#cb2-321" aria-hidden="true" tabindex="-1"></a>  <span class="co">#         hjust = 2, vjust = 1.1)</span></span>
<span id="cb2-322"><a href="#cb2-322" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb2-323"><a href="#cb2-323" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-324"><a href="#cb2-324" aria-hidden="true" tabindex="-1"></a>Beta_plots <span class="ot">=</span> <span class="fu">wrap_plots</span>(<span class="fu">c</span>(Beta_plots1, Beta_plots2),<span class="at">ncol =</span> <span class="dv">3</span>,<span class="at">legend.position =</span> <span class="st">"none"</span>)</span>
<span id="cb2-325"><a href="#cb2-325" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">"Real_Plots/Beta_Diversity.pdf"</span>,Beta_plots,<span class="at">width =</span> <span class="dv">5</span>,<span class="at">height =</span> <span class="fl">3.5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="differential-analysis-and-results-visualization" class="level2">
<h2 class="anchored" data-anchor-id="differential-analysis-and-results-visualization">3. Differential analysis and results visualization</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Univariate Results -----------------------------------------------------------------------</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>physeq <span class="ot">=</span> physeq<span class="fl">.16</span>s.genus</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(physeq<span class="sc">@</span>tax_table) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"Kingdom"</span>, <span class="st">"Phylum"</span>, <span class="st">"Class"</span>,<span class="st">"Order"</span>, <span class="st">"Family"</span>, <span class="st">"Genus"</span>, <span class="st">"Species"</span>)</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>DA_taxa <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"Phylum"</span>,<span class="st">"Class"</span>,<span class="st">"Order"</span>,<span class="st">"Family"</span>,<span class="st">"Genus"</span>)</span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>group <span class="ot">=</span> <span class="st">"diagnosis"</span></span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a>lefse_results <span class="ot">=</span> <span class="fu">run_lefse</span>(physeq,<span class="at">group =</span> group,<span class="at">norm =</span> <span class="st">"CPM"</span>,<span class="at">kw_cutoff=</span><span class="dv">1</span>,<span class="at">wilcoxon_cutoff =</span> <span class="dv">1</span>,<span class="at">lda_cutoff=</span><span class="dv">0</span>,<span class="at">taxa_rank =</span> <span class="st">"Genus"</span>)</span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a>aldex_results <span class="ot">=</span> <span class="fu">run_aldex</span>(physeq,<span class="at">group =</span> group,<span class="at">p_adjust =</span> <span class="st">"BH"</span>,<span class="at">pvalue_cutoff =</span> <span class="dv">1</span>)</span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a>edger_results <span class="ot">=</span> <span class="fu">run_edger</span>(physeq,<span class="at">group =</span> group,<span class="at">p_adjust =</span> <span class="st">"BH"</span>,<span class="at">pvalue_cutoff =</span> <span class="dv">1</span>)</span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a>deseq2_results <span class="ot">=</span> <span class="fu">run_deseq2</span>(physeq,<span class="at">group =</span> group,<span class="at">fitType =</span> <span class="st">"parametric"</span>,<span class="at">sfType =</span> <span class="st">"poscounts"</span>,<span class="at">p_adjust =</span> <span class="st">"BH"</span>,<span class="at">pvalue_cutoff =</span> <span class="dv">1</span>)</span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a>metagenomeseq_results <span class="ot">=</span> <span class="fu">run_metagenomeseq</span>(physeq,<span class="at">group =</span> group,<span class="at">p_adjust =</span> <span class="st">"BH"</span>,<span class="at">pvalue_cutoff =</span> <span class="dv">1</span>,<span class="at">method =</span> <span class="st">"ZIG"</span>)</span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a>limma_results <span class="ot">=</span> <span class="fu">run_limma_voom</span>(physeq,<span class="at">group =</span> group,<span class="at">norm =</span> <span class="st">"none"</span>,<span class="at">p_adjust =</span> <span class="st">"BH"</span>,<span class="at">pvalue_cutoff =</span><span class="dv">1</span>)</span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a>ancombc_results <span class="ot">=</span> <span class="fu">run_ancombc</span>(physeq,<span class="at">group =</span> group,<span class="at">p_adjust =</span> <span class="st">"BH"</span>,<span class="at">pvalue_cutoff =</span><span class="dv">1</span>)</span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-15"><a href="#cb3-15" aria-hidden="true" tabindex="-1"></a>out <span class="ot">=</span> <span class="fu">list</span>(<span class="at">lefse_results =</span> lefse_results,<span class="at">aldex_results =</span> aldex_results,</span>
<span id="cb3-16"><a href="#cb3-16" aria-hidden="true" tabindex="-1"></a>           <span class="at">edger_results =</span> edger_results,<span class="at">deseq2_results =</span> deseq2_results,</span>
<span id="cb3-17"><a href="#cb3-17" aria-hidden="true" tabindex="-1"></a>           <span class="at">metagenomeseq_results =</span> metagenomeseq_results,</span>
<span id="cb3-18"><a href="#cb3-18" aria-hidden="true" tabindex="-1"></a>           <span class="at">limma_results =</span> limma_results,</span>
<span id="cb3-19"><a href="#cb3-19" aria-hidden="true" tabindex="-1"></a>           <span class="at">ancombc_results =</span> ancombc_results)</span>
<span id="cb3-20"><a href="#cb3-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-21"><a href="#cb3-21" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(out,<span class="at">file =</span> <span class="st">"SixS_DA_results.Rdata"</span>)</span>
<span id="cb3-22"><a href="#cb3-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-23"><a href="#cb3-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-24"><a href="#cb3-24" aria-hidden="true" tabindex="-1"></a><span class="do">## Visualization --------------------------------------------------------------------</span></span>
<span id="cb3-25"><a href="#cb3-25" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(reshape2)</span>
<span id="cb3-26"><a href="#cb3-26" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb3-27"><a href="#cb3-27" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyr)</span>
<span id="cb3-28"><a href="#cb3-28" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"WGS_DA_results.Rdata"</span>)</span>
<span id="cb3-29"><a href="#cb3-29" aria-hidden="true" tabindex="-1"></a>DA_WGS <span class="ot">=</span> lefse_results</span>
<span id="cb3-30"><a href="#cb3-30" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="st">"SixS_DA_results.Rdata"</span>)</span>
<span id="cb3-31"><a href="#cb3-31" aria-hidden="true" tabindex="-1"></a>DA_16S <span class="ot">=</span> lefse_results</span>
<span id="cb3-32"><a href="#cb3-32" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="st">"out"</span>)</span>
<span id="cb3-33"><a href="#cb3-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-34"><a href="#cb3-34" aria-hidden="true" tabindex="-1"></a>df_16S <span class="ot">=</span> <span class="fu">as.data.frame</span>(DA_16S<span class="sc">@</span>marker_table)           </span>
<span id="cb3-35"><a href="#cb3-35" aria-hidden="true" tabindex="-1"></a>df_16S<span class="sc">$</span>signed_lda <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(df_16S<span class="sc">$</span>enrich_group <span class="sc">==</span> <span class="st">"uc"</span>, df_16S<span class="sc">$</span>ef_lda, <span class="sc">-</span>df_16S<span class="sc">$</span>ef_lda)</span>
<span id="cb3-36"><a href="#cb3-36" aria-hidden="true" tabindex="-1"></a>df_16S<span class="sc">$</span>significant <span class="ot">&lt;-</span> df_16S<span class="sc">$</span>pvalue <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> df_16S<span class="sc">$</span>ef_lda<span class="sc">&gt;=</span><span class="dv">3</span></span>
<span id="cb3-37"><a href="#cb3-37" aria-hidden="true" tabindex="-1"></a>df_16S<span class="sc">$</span>platform <span class="ot">=</span> <span class="st">"16S"</span></span>
<span id="cb3-38"><a href="#cb3-38" aria-hidden="true" tabindex="-1"></a>matched_idx <span class="ot">&lt;-</span> <span class="fu">match</span>(df_16S<span class="sc">$</span>feature, <span class="fu">rownames</span>(ctax_table_c))</span>
<span id="cb3-39"><a href="#cb3-39" aria-hidden="true" tabindex="-1"></a>df_16S<span class="sc">$</span>feature[<span class="sc">!</span><span class="fu">is.na</span>(matched_idx)] <span class="ot">&lt;-</span> ctax_table_c<span class="sc">$</span>genus[matched_idx[<span class="sc">!</span><span class="fu">is.na</span>(matched_idx)]]</span>
<span id="cb3-40"><a href="#cb3-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-41"><a href="#cb3-41" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb3-42"><a href="#cb3-42" aria-hidden="true" tabindex="-1"></a>df_WGS <span class="ot">=</span> <span class="fu">as.data.frame</span>(DA_WGS<span class="sc">@</span>marker_table)           </span>
<span id="cb3-43"><a href="#cb3-43" aria-hidden="true" tabindex="-1"></a>df_WGS<span class="sc">$</span>signed_lda <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(df_WGS<span class="sc">$</span>enrich_group <span class="sc">==</span> <span class="st">"uc"</span>, df_WGS<span class="sc">$</span>ef_lda, <span class="sc">-</span>df_WGS<span class="sc">$</span>ef_lda)</span>
<span id="cb3-44"><a href="#cb3-44" aria-hidden="true" tabindex="-1"></a>df_WGS<span class="sc">$</span>significant <span class="ot">&lt;-</span> df_WGS<span class="sc">$</span>pvalue <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> df_WGS<span class="sc">$</span>ef_lda <span class="sc">&gt;=</span> <span class="dv">3</span></span>
<span id="cb3-45"><a href="#cb3-45" aria-hidden="true" tabindex="-1"></a>df_WGS<span class="sc">$</span>platform <span class="ot">=</span> <span class="st">"WGS"</span></span>
<span id="cb3-46"><a href="#cb3-46" aria-hidden="true" tabindex="-1"></a>matched_idx <span class="ot">&lt;-</span> <span class="fu">match</span>(df_WGS<span class="sc">$</span>feature, <span class="fu">rownames</span>(ctax_table_c1))</span>
<span id="cb3-47"><a href="#cb3-47" aria-hidden="true" tabindex="-1"></a>df_WGS<span class="sc">$</span>feature[<span class="sc">!</span><span class="fu">is.na</span>(matched_idx)] <span class="ot">&lt;-</span> ctax_table_c1<span class="sc">$</span>genus[matched_idx[<span class="sc">!</span><span class="fu">is.na</span>(matched_idx)]]</span>
<span id="cb3-48"><a href="#cb3-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-49"><a href="#cb3-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-50"><a href="#cb3-50" aria-hidden="true" tabindex="-1"></a>sig_16S <span class="ot">&lt;-</span> df_16S<span class="sc">$</span>feature[df_16S<span class="sc">$</span>significant]</span>
<span id="cb3-51"><a href="#cb3-51" aria-hidden="true" tabindex="-1"></a>sig_16S <span class="ot">&lt;-</span> ctax_table_c<span class="sc">$</span>genus</span>
<span id="cb3-52"><a href="#cb3-52" aria-hidden="true" tabindex="-1"></a>sig_WGS <span class="ot">&lt;-</span> df_WGS<span class="sc">$</span>feature[df_WGS<span class="sc">$</span>significant]</span>
<span id="cb3-53"><a href="#cb3-53" aria-hidden="true" tabindex="-1"></a>sig_WGS <span class="ot">&lt;-</span> ctax_table_c1<span class="sc">$</span>genus</span>
<span id="cb3-54"><a href="#cb3-54" aria-hidden="true" tabindex="-1"></a>all_common <span class="ot">&lt;-</span> <span class="fu">intersect</span>(sig_16S,sig_WGS)</span>
<span id="cb3-55"><a href="#cb3-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-56"><a href="#cb3-56" aria-hidden="true" tabindex="-1"></a><span class="co"># Replace p__Firmicutes with p__Bacillota</span></span>
<span id="cb3-57"><a href="#cb3-57" aria-hidden="true" tabindex="-1"></a>df_WGS<span class="sc">$</span>feature <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"p__Firmicutes"</span>, <span class="st">"p__Bacillota"</span>, df_WGS<span class="sc">$</span>feature)</span>
<span id="cb3-58"><a href="#cb3-58" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-59"><a href="#cb3-59" aria-hidden="true" tabindex="-1"></a><span class="co"># Replace p__Proteobacteria with p__Pseudomonadota</span></span>
<span id="cb3-60"><a href="#cb3-60" aria-hidden="true" tabindex="-1"></a>df_WGS<span class="sc">$</span>feature <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"p__Proteobacteria"</span>, <span class="st">"p__Pseudomonadota"</span>, df_WGS<span class="sc">$</span>feature)</span>
<span id="cb3-61"><a href="#cb3-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-62"><a href="#cb3-62" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-63"><a href="#cb3-63" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">rbind</span>(df_16S, df_WGS) <span class="sc">%&gt;%</span></span>
<span id="cb3-64"><a href="#cb3-64" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(significant) <span class="sc">%&gt;%</span></span>
<span id="cb3-65"><a href="#cb3-65" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb3-66"><a href="#cb3-66" aria-hidden="true" tabindex="-1"></a>    <span class="at">group =</span> <span class="fu">case_when</span>(</span>
<span id="cb3-67"><a href="#cb3-67" aria-hidden="true" tabindex="-1"></a>      <span class="sc">!</span>(feature <span class="sc">%in%</span> all_common) <span class="sc">&amp;</span> platform <span class="sc">==</span> <span class="st">"16S"</span> <span class="sc">~</span> <span class="st">"Unique-16S"</span>,</span>
<span id="cb3-68"><a href="#cb3-68" aria-hidden="true" tabindex="-1"></a>      <span class="sc">!</span>(feature <span class="sc">%in%</span> all_common) <span class="sc">&amp;</span> platform <span class="sc">==</span> <span class="st">"WGS"</span> <span class="sc">~</span> <span class="st">"Unique-WGS"</span>,</span>
<span id="cb3-69"><a href="#cb3-69" aria-hidden="true" tabindex="-1"></a>      feature <span class="sc">%in%</span> all_common <span class="sc">&amp;</span> feature <span class="sc">%in%</span> sig_16S <span class="sc">&amp;</span> <span class="sc">!</span>(feature <span class="sc">%in%</span> sig_WGS) <span class="sc">~</span> <span class="st">"Common-16S"</span>,</span>
<span id="cb3-70"><a href="#cb3-70" aria-hidden="true" tabindex="-1"></a>      feature <span class="sc">%in%</span> all_common <span class="sc">&amp;</span> feature <span class="sc">%in%</span> sig_WGS <span class="sc">&amp;</span> <span class="sc">!</span>(feature <span class="sc">%in%</span> sig_16S) <span class="sc">~</span> <span class="st">"Common-WGS"</span>,</span>
<span id="cb3-71"><a href="#cb3-71" aria-hidden="true" tabindex="-1"></a>      feature <span class="sc">%in%</span> all_common <span class="sc">&amp;</span> feature <span class="sc">%in%</span> sig_16S <span class="sc">&amp;</span> feature <span class="sc">%in%</span> sig_WGS <span class="sc">~</span> <span class="st">"Common-both"</span></span>
<span id="cb3-72"><a href="#cb3-72" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb3-73"><a href="#cb3-73" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb3-74"><a href="#cb3-74" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-75"><a href="#cb3-75" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-76"><a href="#cb3-76" aria-hidden="true" tabindex="-1"></a>df<span class="sc">$</span>group <span class="ot">&lt;-</span> <span class="fu">factor</span>(df<span class="sc">$</span>group, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"Unique-16S"</span>, <span class="st">"Common-16S"</span>, <span class="st">"Common-both"</span>, <span class="st">"Common-WGS"</span>, <span class="st">"Unique-WGS"</span>))</span>
<span id="cb3-77"><a href="#cb3-77" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-78"><a href="#cb3-78" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> df <span class="sc">%&gt;%</span></span>
<span id="cb3-79"><a href="#cb3-79" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(group, <span class="fu">desc</span>(signed_lda)) <span class="sc">%&gt;%</span></span>
<span id="cb3-80"><a href="#cb3-80" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">feature =</span> <span class="fu">factor</span>(feature, <span class="at">levels =</span> <span class="fu">rev</span>(<span class="fu">unique</span>(feature))))</span>
<span id="cb3-81"><a href="#cb3-81" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-82"><a href="#cb3-82" aria-hidden="true" tabindex="-1"></a>df<span class="sc">$</span>feature <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"</span><span class="sc">\\</span><span class="st">|"</span>, <span class="st">";"</span>, df<span class="sc">$</span>feature)</span>
<span id="cb3-83"><a href="#cb3-83" aria-hidden="true" tabindex="-1"></a>df<span class="sc">$</span>feature <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"k__Bacteria;</span><span class="sc">\\</span><span class="st">|?"</span>, <span class="st">""</span>, df<span class="sc">$</span>feature)</span>
<span id="cb3-84"><a href="#cb3-84" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-85"><a href="#cb3-85" aria-hidden="true" tabindex="-1"></a>plot <span class="ot">=</span> <span class="fu">ggplot</span>(df, <span class="fu">aes</span>(<span class="at">x =</span> signed_lda, <span class="at">y =</span> feature, <span class="at">fill =</span> platform)) <span class="sc">+</span></span>
<span id="cb3-86"><a href="#cb3-86" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">width =</span> <span class="fl">0.6</span>, <span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="at">width =</span> <span class="fl">0.7</span>)) <span class="sc">+</span></span>
<span id="cb3-87"><a href="#cb3-87" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">0</span>, <span class="at">color =</span> <span class="st">"gray40"</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb3-88"><a href="#cb3-88" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(</span>
<span id="cb3-89"><a href="#cb3-89" aria-hidden="true" tabindex="-1"></a>    <span class="at">values =</span> <span class="fu">c</span>(<span class="st">"16S"</span> <span class="ot">=</span> <span class="st">"#ffa551"</span>, <span class="st">"WGS"</span> <span class="ot">=</span> <span class="st">"#70afdf"</span>)</span>
<span id="cb3-90"><a href="#cb3-90" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb3-91"><a href="#cb3-91" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlim</span>(<span class="sc">-</span><span class="dv">5</span>,<span class="dv">5</span>)<span class="sc">+</span></span>
<span id="cb3-92"><a href="#cb3-92" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="cn">NULL</span>, <span class="at">y =</span> <span class="cn">NULL</span>, <span class="at">fill =</span> <span class="st">"Platform"</span>, <span class="at">title =</span> <span class="cn">NULL</span>) <span class="sc">+</span></span>
<span id="cb3-93"><a href="#cb3-93" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>()<span class="sc">+</span></span>
<span id="cb3-94"><a href="#cb3-94" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb3-95"><a href="#cb3-95" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">15</span>),</span>
<span id="cb3-96"><a href="#cb3-96" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.position =</span> <span class="st">"none"</span>,  <span class="co"># Top right inside plot</span></span>
<span id="cb3-97"><a href="#cb3-97" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.justification =</span> <span class="fu">c</span>(<span class="st">"right"</span>, <span class="st">"top"</span>),</span>
<span id="cb3-98"><a href="#cb3-98" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="st">"white"</span>, <span class="at">color =</span> <span class="st">"gray80"</span>),</span>
<span id="cb3-99"><a href="#cb3-99" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="st">"transparent"</span>, <span class="at">color =</span> <span class="cn">NA</span>),</span>
<span id="cb3-100"><a href="#cb3-100" aria-hidden="true" tabindex="-1"></a>    <span class="at">plot.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="st">"transparent"</span>, <span class="at">color =</span> <span class="cn">NA</span>)</span>
<span id="cb3-101"><a href="#cb3-101" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>